[BUG] `test_hash_groupby_collect_set_on_nested_type` and `test_hash_reduction_collect_set_on_nested_type` failed

**Describe the bug**
`test_hash_groupby_collect_set_on_nested_type` and `test_hash_reduction_collect_set_on_nested_type` will failed after this commit: https://github.com/NVIDIA/spark-rapids/commit/615156a2e276bf8ae553ef7110dafc37aefacda0, which changed the `DataGen` in integration tests. It also marked the failed cases as XFAIL.

This failure should not be caused by this commit. Before this commit, simply changing the seed (like 4) of the data generator in the test can make them fail.

**Steps/Code to reproduce bug**
```python
@ignore_order(local=True)
@pytest.mark.parametrize('data_gen', _gen_data_for_collect_set_op, ids=idfn)
def test_hash_groupby_collect_set_on_nested_type(data_gen):
    assert_gpu_and_cpu_are_equal_collect(
        lambda spark: gen_df(spark, data_gen, length=100)
            .groupby('a')
            .agg(f.sort_array(f.collect_set('b'))))
```
and
```python
@ignore_order(local=True)
@pytest.mark.parametrize('data_gen', _gen_data_for_collect_set_op, ids=idfn)
def test_hash_reduction_collect_set_on_nested_type(data_gen):
    assert_gpu_and_cpu_are_equal_collect(
        lambda spark: gen_df(spark, data_gen, length=100)
            .agg(f.sort_array(f.collect_set('b'))))
```

got:
```
FAILED ../../src/main/python/hash_aggregate_test.py::test_hash_groupby_collect_set_on_nested_type[[('a', RepeatSeq(Long)), ('b', RepeatSeq(Struct(['child0', Byte],['child1', Short],['child2', Integer],['child3', Long],['child4', Float],['child5', Double],['child6', String],['child7', Boolean],['child8', Date],['child9', Timestamp],['child10', Null])))]][INJECT_OOM, IGNORE_ORDER({'local': True})]
FAILED ../../src/main/python/hash_aggregate_test.py::test_hash_reduction_collect_set_on_nested_type[[('a', RepeatSeq(Long)), ('b', RepeatSeq(Struct(['child0', Byte],['child1', Short],['child2', Integer],['child3', Long],['child4', Float],['child5', Double],['child6', String],['child7', Boolean],['child8', Date],['child9', Timestamp],['child10', Null])))]][INJECT_OOM, IGNORE_ORDER({'local': True})]
```

log:
```
================================================================= FAILURES ==================================================================
_ test_hash_groupby_collect_set_on_nested_type[[('a', RepeatSeq(Long)), ('b', RepeatSeq(Struct(['child0', Byte],['child1', Short],['child2', Integer],['child3', Long],['child4', Float],['child5', Double],['child6', String],['child7', Boolean],['child8', Date],['child9', Timestamp],['child10', Null])))]] _
[gw0] linux -- Python 3.8.3 /home/haoyangl/.pyenv/versions/3.8.3/bin/python

data_gen = [('a', RepeatSeq(Long)), ('b', RepeatSeq(Struct(['child0', Byte],['child1', Short],['child2', Integer],['child3', Long...,['child5', Double],['child6', String],['child7', Boolean],['child8', Date],['child9', Timestamp],['child10', Null])))]

    @ignore_order(local=True)
    @pytest.mark.parametrize('data_gen', _gen_data_for_collect_set_op, ids=idfn)
    # @pytest.mark.xfail(condition=is_before_spark_330(), reason='https://github.com/NVIDIA/spark-rapids/issues/8716')
    def test_hash_groupby_collect_set_on_nested_type(data_gen):
>       assert_gpu_and_cpu_are_equal_collect(
            lambda spark: gen_df(spark, data_gen, length=100)
                .groupby('a')
                .agg(f.sort_array(f.collect_set('b'))))

../../src/main/python/hash_aggregate_test.py:693:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../src/main/python/asserts.py:561: in assert_gpu_and_cpu_are_equal_collect
    _assert_gpu_and_cpu_are_equal(func, 'COLLECT', conf=conf, is_cpu_first=is_cpu_first)
../../src/main/python/asserts.py:492: in _assert_gpu_and_cpu_are_equal
    assert_equal(from_cpu, from_gpu)
../../src/main/python/asserts.py:105: in assert_equal
    _assert_equal(cpu, gpu, float_check=get_float_check(), path=[])
../../src/main/python/asserts.py:41: in _assert_equal
    _assert_equal(cpu[index], gpu[index], float_check, path + [index])
../../src/main/python/asserts.py:34: in _assert_equal
    _assert_equal(cpu[field], gpu[field], float_check, path + [field])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cpu = [Row(child0=-50, child1=-31138, child2=-1807864316, child3=-4041992126494898742, child4=4.065499056633676e+31, child5=...hild7=False, child8=datetime.date(8508, 12, 19), child9=datetime.datetime(7713, 4, 6, 2, 18, 9, 495678), child10=None)]
gpu = [Row(child0=-50, child1=-31138, child2=-1807864316, child3=-4041992126494898742, child4=4.065499056633676e+31, child5=...hild7=False, child8=datetime.date(8508, 12, 19), child9=datetime.datetime(7713, 4, 6, 2, 18, 9, 495678), child10=None)]
float_check = <function get_float_check.<locals>.<lambda> at 0x7fcebc7fb280>, path = [3, 'sort_array(collect_set(b), true)']

    def _assert_equal(cpu, gpu, float_check, path):
        t = type(cpu)
        if (t is Row):
            assert len(cpu) == len(gpu), "CPU and GPU row have different lengths at {} CPU: {} GPU: {}".format(path, len(cpu), len(gpu))
            if hasattr(cpu, "__fields__") and hasattr(gpu, "__fields__"):
                assert cpu.__fields__ == gpu.__fields__, "CPU and GPU row have different fields at {} CPU: {} GPU: {}".format(path, cpu.__fields__, gpu.__fields__)
                for field in cpu.__fields__:
                    _assert_equal(cpu[field], gpu[field], float_check, path + [field])
            else:
                for index in range(len(cpu)):
                    _assert_equal(cpu[index], gpu[index], float_check, path + [index])
        elif (t is list):
>           assert len(cpu) == len(gpu), "CPU and GPU list have different lengths at {} CPU: {} GPU: {}".format(path, len(cpu), len(gpu))
E           AssertionError: CPU and GPU list have different lengths at [3, 'sort_array(collect_set(b), true)'] CPU: 3 GPU: 4

../../src/main/python/asserts.py:39: AssertionError
----------------------------------------------------------- Captured stdout call ------------------------------------------------------------
### CPU RUN ###
### GPU RUN ###
### COLLECT: GPU TOOK 1.3745741844177246 CPU TOOK 3.6840076446533203 ###
CPU OUTPUT: [Row(a=-7540734677356764604, sort_array(collect_set(b), true)=[Row(child0=-110, child1=-11291, child2=358420087, child3=894377461322806935, child4=-1.6834864679694056e-31, child5=1.0303888663388996e-26, child6='¯óÏôÅkùê,dÌA~|×AÖ\x08VJé\x08yX^i{.\x1b\x83', child7=False, child8=datetime.date(4899, 3, 2), child9=datetime.datetime(6861, 9, 7, 19, 16, 50, 365146), child10=None), Row(child0=-61, child1=-18968, child2=639916889, child3=2251193521986836658, child4=1.7728260685861134e+25, child5=-1.6653918844116466e+240, child6=' 7jgv#\x1aÑÚªï\x8d\x9fð\x15\x83\x1fîìAæöl\x0bå\\\x90æ9â', child7=True, child8=datetime.date(353, 7, 23), child9=datetime.datetime(551, 6, 24, 14, 12, 58, 187381), child10=None), Row(child0=-41, child1=-24617, child2=848988934, child3=-7858370133784586516, child4=-1.740257146253725e-20, child5=3.5575370470599927e+284, child6='õ^\x1f·\x97ú·ÖF{/ZR*ø\x7fCýö\x05%A1Ð¶d\x04\x89øx', child7=True, child8=datetime.date(7802, 5, 13), child9=datetime.datetime(1279, 10, 18, 15, 49, 5, 515554), child10=None)]), Row(a=-5831592707909023540, sort_array(collect_set(b), true)=[Row(child0=-36, child1=28459, child2=2096953985, child3=1287002480498025462, child4=963.7640991210938, child5=9.376528689861087e+117, child6='\x05Â,ÀøYªe$«q;~\x04ëâ\x13h\x98ÇR\x04\x1e\x00©4@-\x0b°', child7=False, child8=datetime.date(8000, 2, 29), child9=datetime.datetime(5690, 4, 15, 13, 26, 32, 4190), child10=None), Row(child0=1, child1=-23550, child2=823668003, child3=-6580116972850632313, child4=3.816635312018178e+23, child5=8.600719030180411e+134, child6='\x0eM!R\x02ÇB\x1a¡^õ\x8cCÎµ\x1dpØX-Ùrz\x07\x9c¨\x1cÅ¨¡', child7=None, child8=datetime.date(5437, 1, 2), child9=None, child10=None)]), Row(a=-5133656973475552689, sort_array(collect_set(b), true)=[Row(child0=-108, child1=1267, child2=1917713880, child3=None, child4=None, child5=4.3214483342777574e-117, child6='[H\x89µ³hß\x14Æ\x12_XÕµoy\x08Y£àK;uk\x06\x15ç\x18\x02#', child7=True, child8=datetime.date(1590, 12, 27), child9=datetime.datetime(6166, 6, 5, 15, 21, 41, 824645), child10=None), Row(child0=-87, child1=18519, child2=105544272, child3=6446878210185090927, child4=None, child5=-4.2132496663975606e+247, child6='¦\r\x15\x16ep[|p\x9aË\x17ZZû\x82\x86\rê¼¨Ð³AúÍÿ\tÀò', child7=False, child8=datetime.date(3851, 2, 13), child9=datetime.datetime(297, 2, 6, 14, 13, 25, 35309), child10=None), Row(child0=6, child1=None, child2=-394822741, child3=-6874717541235784212, child4=4243308150784.0, child5=-3.638751059169508e+285, child6='M\x80²ÝÄ\x81¬m]ø\x14å\x05L·\x99®\x8e\x9a\x1a}oÝ\x01à\nãÔ\x84\x08', child7=False, child8=datetime.date(7026, 5, 5), child9=datetime.datetime(860, 3, 3, 14, 32, 46, 946330), child10=None)]), Row(a=-4426181692283497353, sort_array(collect_set(b), true)=[Row(child0=-50, child1=-31138, child2=-1807864316, child3=-4041992126494898742, child4=4.065499056633676e+31, child5=7.554028173936504e-95, child6='ÆªOo©\x85·2Ôo!áP\x94\x9eþädÚ\x90õ4¢=L\x9däVi}', child7=True, child8=datetime.date(18, 7, 4), child9=datetime.datetime(6109, 8, 17, 22, 8, 3, 694767), child10=None), Row(child0=-27, child1=-24753, child2=-1726248225, child3=3307548583936611830, child4=-3.656386885506302e+18, child5=nan, child6='5\x92%w|ð\x9f\x98!\x887DÚdÌ$\x95Xò\xad\x98_ÿ>\x0b¡\tÇ(µ', child7=False, child8=datetime.date(9497, 10, 18), child9=datetime.datetime(8913, 8, 27, 22, 48, 29, 877849), child10=None), Row(child0=39, child1=-22177, child2=347245469, child3=5615010007311415123, child4=-0.16667649149894714, child5=-2.1002678349193272e+151, child6="%{³ÁaÃ\x13ò£|dÐ'Y\x04B½u\x99èÈ(!Ú\x13,ÜhÔþ", child7=False, child8=datetime.date(8508, 12, 19), child9=datetime.datetime(7713, 4, 6, 2, 18, 9, 495678), child10=None)]), Row(a=-3917032101531217289, sort_array(collect_set(b), true)=[Row(child0=-108, child1=1267, child2=1917713880, child3=None, child4=None, child5=4.3214483342777574e-117, child6='[H\x89µ³hß\x14Æ\x12_XÕµoy\x08Y£àK;uk\x06\x15ç\x18\x02#', child7=True, child8=datetime.date(1590, 12, 27), child9=datetime.datetime(6166, 6, 5, 15, 21, 41, 824645), child10=None), Row(child0=-87, child1=18519, child2=105544272, child3=6446878210185090927, child4=None, child5=-4.2132496663975606e+247, child6='¦\r\x15\x16ep[|p\x9aË\x17ZZû\x82\x86\rê¼¨Ð³AúÍÿ\tÀò', child7=False, child8=datetime.date(3851, 2, 13), child9=datetime.datetime(297, 2, 6, 14, 13, 25, 35309), child10=None), Row(child0=6, child1=None, child2=-394822741, child3=-6874717541235784212, child4=4243308150784.0, child5=-3.638751059169508e+285, child6='M\x80²ÝÄ\x81¬m]ø\x14å\x05L·\x99®\x8e\x9a\x1a}oÝ\x01à\nãÔ\x84\x08', child7=False, child8=datetime.date(7026, 5, 5), child9=datetime.datetime(860, 3, 3, 14, 32, 46, 946330), child10=None)]), Row(a=-3502159106106506455, sort_array(collect_set(b), true)=[Row(child0=-108, child1=1267, child2=1917713880, child3=None, child4=None, child5=4.3214483342777574e-117, child6='[H\x89µ³hß\x14Æ\x12_XÕµoy\x08Y£àK;uk\x06\x15ç\x18\x02#', child7=True, child8=datetime.date(1590, 12, 27), child9=datetime.datetime(6166, 6, 5, 15, 21, 41, 824645), child10=None), Row(child0=-87, child1=18519, child2=105544272, child3=6446878210185090927, child4=None, child5=-4.2132496663975606e+247, child6='¦\r\x15\x16ep[|p\x9aË\x17ZZû\x82\x86\rê¼¨Ð³AúÍÿ\tÀò', child7=False, child8=datetime.date(3851, 2, 13), child9=datetime.datetime(297, 2, 6, 14, 13, 25, 35309), child10=None), Row(child0=6, child1=None, child2=-394822741, child3=-6874717541235784212, child4=4243308150784.0, child5=-3.638751059169508e+285, child6='M\x80²ÝÄ\x81¬m]ø\x14å\x05L·\x99®\x8e\x9a\x1a}oÝ\x01à\nãÔ\x84\x08', child7=False, child8=datetime.date(7026, 5, 5), child9=datetime.datetime(860, 3, 3, 14, 32, 46, 946330), child10=None)]), Row(a=-2697073954890740236, sort_array(collect_set(b), true)=[Row(child0=-110, child1=-11291, child2=358420087, child3=894377461322806935, child4=-1.6834864679694056e-31, child5=1.0303888663388996e-26, child6='¯óÏôÅkùê,dÌA~|×AÖ\x08VJé\x08yX^i{.\x1b\x83', child7=False, child8=datetime.date(4899, 3, 2), child9=datetime.datetime(6861, 9, 7, 19, 16, 50, 365146), child10=None), Row(child0=-61, child1=-18968, child2=639916889, child3=2251193521986836658, child4=1.7728260685861134e+25, child5=-1.6653918844116466e+240, child6=' 7jgv#\x1aÑÚªï\x8d\x9fð\x15\x83\x1fîìAæöl\x0bå\\\x90æ9â', child7=True, child8=datetime.date(353, 7, 23), child9=datetime.datetime(551, 6, 24, 14, 12, 58, 187381), child10=None), Row(child0=-41, child1=-24617, child2=848988934, child3=-7858370133784586516, child4=-1.740257146253725e-20, child5=3.5575370470599927e+284, child6='õ^\x1f·\x97ú·ÖF{/ZR*ø\x7fCýö\x05%A1Ð¶d\x04\x89øx', child7=True, child8=datetime.date(7802, 5, 13), child9=datetime.datetime(1279, 10, 18, 15, 49, 5, 515554), child10=None)]), Row(a=-2123199122092230623, sort_array(collect_set(b), true)=[Row(child0=-110, child1=-11291, child2=358420087, child3=894377461322806935, child4=-1.6834864679694056e-31, child5=1.0303888663388996e-26, child6='¯óÏôÅkùê,dÌA~|×AÖ\x08VJé\x08yX^i{.\x1b\x83', child7=False, child8=datetime.date(4899, 3, 2), child9=datetime.datetime(6861, 9, 7, 19, 16, 50, 365146), child10=None), Row(child0=-61, child1=-18968, child2=639916889, child3=2251193521986836658, child4=1.7728260685861134e+25, child5=-1.6653918844116466e+240, child6=' 7jgv#\x1aÑÚªï\x8d\x9fð\x15\x83\x1fîìAæöl\x0bå\\\x90æ9â', child7=True, child8=datetime.date(353, 7, 23), child9=datetime.datetime(551, 6, 24, 14, 12, 58, 187381), child10=None), Row(child0=-41, child1=-24617, child2=848988934, child3=-7858370133784586516, child4=-1.740257146253725e-20, child5=3.5575370470599927e+284, child6='õ^\x1f·\x97ú·ÖF{/ZR*ø\x7fCýö\x05%A1Ð¶d\x04\x89øx', child7=True, child8=datetime.date(7802, 5, 13), child9=datetime.datetime(1279, 10, 18, 15, 49, 5, 515554), child10=None)]), Row(a=-1, sort_array(collect_set(b), true)=[Row(child0=-36, child1=28459, child2=2096953985, child3=1287002480498025462, child4=963.7640991210938, child5=9.376528689861087e+117, child6='\x05Â,ÀøYªe$«q;~\x04ëâ\x13h\x98ÇR\x04\x1e\x00©4@-\x0b°', child7=False, child8=datetime.date(8000, 2, 29), child9=datetime.datetime(5690, 4, 15, 13, 26, 32, 4190), child10=None), Row(child0=1, child1=-23550, child2=823668003, child3=-6580116972850632313, child4=3.816635312018178e+23, child5=8.600719030180411e+134, child6='\x0eM!R\x02ÇB\x1a¡^õ\x8cCÎµ\x1dpØX-Ùrz\x07\x9c¨\x1cÅ¨¡', child7=None, child8=datetime.date(5437, 1, 2), child9=None, child10=None)]), Row(a=207981845540287738, sort_array(collect_set(b), true)=[Row(child0=-110, child1=-11291, child2=358420087, child3=894377461322806935, child4=-1.6834864679694056e-31, child5=1.0303888663388996e-26, child6='¯óÏôÅkùê,dÌA~|×AÖ\x08VJé\x08yX^i{.\x1b\x83', child7=False, child8=datetime.date(4899, 3, 2), child9=datetime.datetime(6861, 9, 7, 19, 16, 50, 365146), child10=None), Row(child0=-61, child1=-18968, child2=639916889, child3=2251193521986836658, child4=1.7728260685861134e+25, child5=-1.6653918844116466e+240, child6=' 7jgv#\x1aÑÚªï\x8d\x9fð\x15\x83\x1fîìAæöl\x0bå\\\x90æ9â', child7=True, child8=datetime.date(353, 7, 23), child9=datetime.datetime(551, 6, 24, 14, 12, 58, 187381), child10=None), Row(child0=-41, child1=-24617, child2=848988934, child3=-7858370133784586516, child4=-1.740257146253725e-20, child5=3.5575370470599927e+284, child6='õ^\x1f·\x97ú·ÖF{/ZR*ø\x7fCýö\x05%A1Ð¶d\x04\x89øx', child7=True, child8=datetime.date(7802, 5, 13), child9=datetime.datetime(1279, 10, 18, 15, 49, 5, 515554), child10=None)]), Row(a=393905103838704542, sort_array(collect_set(b), true)=[Row(child0=61, child1=-25185, child2=-2111287335, child3=-9223372036854775808, child4=None, child5=4.813757617752221e+57, child6=':\x9aLno»7þöµ\x00ý¦\x98vFWÊ\x9eeslr÷t±²ñ\x1eú', child7=True, child8=datetime.date(6996, 5, 13), child9=datetime.datetime(1691, 9, 13, 20, 49, 32, 239246), child10=None), Row(child0=84, child1=22049, child2=897674100, child3=7081246721334085708, child4=-5.445810266389943e+33, child5=-6.542279108216022e+206, child6='çìéô\x1cÀFS¥`\x04\x97ª¶\x14Ó\r¼\xa0¬ôÉe\x8dá~ì:£\x15', child7=False, child8=None, child9=datetime.datetime(2103, 9, 30, 11, 10, 53, 134710), child10=None), Row(child0=125, child1=9497, child2=2139679874, child3=7956328026836954420, child4=-243864059904.0, child5=2.1874890847202007e-206, child6='w\x96\x96\x1d\x8dÏ\x9b\x80\x86Úcy@5øäP\x86Ê\x81\x9coØr)\x8c{rY\x0c', child7=True, child8=datetime.date(3, 10, 7), child9=datetime.datetime(9259, 12, 8, 20, 39, 28, 751490), child10=None)]), Row(a=875130347651831881, sort_array(collect_set(b), true)=[Row(child0=61, child1=-25185, child2=-2111287335, child3=-9223372036854775808, child4=None, child5=4.813757617752221e+57, child6=':\x9aLno»7þöµ\x00ý¦\x98vFWÊ\x9eeslr÷t±²ñ\x1eú', child7=True, child8=datetime.date(6996, 5, 13), child9=datetime.datetime(1691, 9, 13, 20, 49, 32, 239246), child10=None), Row(child0=84, child1=22049, child2=897674100, child3=7081246721334085708, child4=-5.445810266389943e+33, child5=-6.542279108216022e+206, child6='çìéô\x1cÀFS¥`\x04\x97ª¶\x14Ó\r¼\xa0¬ôÉe\x8dá~ì:£\x15', child7=False, child8=None, child9=datetime.datetime(2103, 9, 30, 11, 10, 53, 134710), child10=None), Row(child0=125, child1=9497, child2=2139679874, child3=7956328026836954420, child4=-243864059904.0, child5=2.1874890847202007e-206, child6='w\x96\x96\x1d\x8dÏ\x9b\x80\x86Úcy@5øäP\x86Ê\x81\x9coØr)\x8c{rY\x0c', child7=True, child8=datetime.date(3, 10, 7), child9=datetime.datetime(9259, 12, 8, 20, 39, 28, 751490), child10=None)]), Row(a=4751953708995107450, sort_array(collect_set(b), true)=[Row(child0=61, child1=-25185, child2=-2111287335, child3=-9223372036854775808, child4=None, child5=4.813757617752221e+57, child6=':\x9aLno»7þöµ\x00ý¦\x98vFWÊ\x9eeslr÷t±²ñ\x1eú', child7=True, child8=datetime.date(6996, 5, 13), child9=datetime.datetime(1691, 9, 13, 20, 49, 32, 239246), child10=None), Row(child0=84, child1=22049, child2=897674100, child3=7081246721334085708, child4=-5.445810266389943e+33, child5=-6.542279108216022e+206, child6='çìéô\x1cÀFS¥`\x04\x97ª¶\x14Ó\r¼\xa0¬ôÉe\x8dá~ì:£\x15', child7=False, child8=None, child9=datetime.datetime(2103, 9, 30, 11, 10, 53, 134710), child10=None), Row(child0=125, child1=9497, child2=2139679874, child3=7956328026836954420, child4=-243864059904.0, child5=2.1874890847202007e-206, child6='w\x96\x96\x1d\x8dÏ\x9b\x80\x86Úcy@5øäP\x86Ê\x81\x9coØr)\x8c{rY\x0c', child7=True, child8=datetime.date(3, 10, 7), child9=datetime.datetime(9259, 12, 8, 20, 39, 28, 751490), child10=None)]), Row(a=6084712057446794809, sort_array(collect_set(b), true)=[Row(child0=-36, child1=28459, child2=2096953985, child3=1287002480498025462, child4=963.7640991210938, child5=9.376528689861087e+117, child6='\x05Â,ÀøYªe$«q;~\x04ëâ\x13h\x98ÇR\x04\x1e\x00©4@-\x0b°', child7=False, child8=datetime.date(8000, 2, 29), child9=datetime.datetime(5690, 4, 15, 13, 26, 32, 4190), child10=None), Row(child0=1, child1=-23550, child2=823668003, child3=-6580116972850632313, child4=3.816635312018178e+23, child5=8.600719030180411e+134, child6='\x0eM!R\x02ÇB\x1a¡^õ\x8cCÎµ\x1dpØX-Ùrz\x07\x9c¨\x1cÅ¨¡', child7=None, child8=datetime.date(5437, 1, 2), child9=None, child10=None)]), Row(a=7198729688045931692, sort_array(collect_set(b), true)=[Row(child0=-108, child1=1267, child2=1917713880, child3=None, child4=None, child5=4.3214483342777574e-117, child6='[H\x89µ³hß\x14Æ\x12_XÕµoy\x08Y£àK;uk\x06\x15ç\x18\x02#', child7=True, child8=datetime.date(1590, 12, 27), child9=datetime.datetime(6166, 6, 5, 15, 21, 41, 824645), child10=None), Row(child0=-87, child1=18519, child2=105544272, child3=6446878210185090927, child4=None, child5=-4.2132496663975606e+247, child6='¦\r\x15\x16ep[|p\x9aË\x17ZZû\x82\x86\rê¼¨Ð³AúÍÿ\tÀò', child7=False, child8=datetime.date(3851, 2, 13), child9=datetime.datetime(297, 2, 6, 14, 13, 25, 35309), child10=None), Row(child0=6, child1=None, child2=-394822741, child3=-6874717541235784212, child4=4243308150784.0, child5=-3.638751059169508e+285, child6='M\x80²ÝÄ\x81¬m]ø\x14å\x05L·\x99®\x8e\x9a\x1a}oÝ\x01à\nãÔ\x84\x08', child7=False, child8=datetime.date(7026, 5, 5), child9=datetime.datetime(860, 3, 3, 14, 32, 46, 946330), child10=None)]), Row(a=7528354001793048440, sort_array(collect_set(b), true)=[Row(child0=61, child1=-25185, child2=-2111287335, child3=-9223372036854775808, child4=None, child5=4.813757617752221e+57, child6=':\x9aLno»7þöµ\x00ý¦\x98vFWÊ\x9eeslr÷t±²ñ\x1eú', child7=True, child8=datetime.date(6996, 5, 13), child9=datetime.datetime(1691, 9, 13, 20, 49, 32, 239246), child10=None), Row(child0=84, child1=22049, child2=897674100, child3=7081246721334085708, child4=-5.445810266389943e+33, child5=-6.542279108216022e+206, child6='çìéô\x1cÀFS¥`\x04\x97ª¶\x14Ó\r¼\xa0¬ôÉe\x8dá~ì:£\x15', child7=False, child8=None, child9=datetime.datetime(2103, 9, 30, 11, 10, 53, 134710), child10=None), Row(child0=125, child1=9497, child2=2139679874, child3=7956328026836954420, child4=-243864059904.0, child5=2.1874890847202007e-206, child6='w\x96\x96\x1d\x8dÏ\x9b\x80\x86Úcy@5øäP\x86Ê\x81\x9coØr)\x8c{rY\x0c', child7=True, child8=datetime.date(3, 10, 7), child9=datetime.datetime(9259, 12, 8, 20, 39, 28, 751490), child10=None)]), Row(a=7618709293599214015, sort_array(collect_set(b), true)=[Row(child0=-36, child1=28459, child2=2096953985, child3=1287002480498025462, child4=963.7640991210938, child5=9.376528689861087e+117, child6='\x05Â,ÀøYªe$«q;~\x04ëâ\x13h\x98ÇR\x04\x1e\x00©4@-\x0b°', child7=False, child8=datetime.date(8000, 2, 29), child9=datetime.datetime(5690, 4, 15, 13, 26, 32, 4190), child10=None), Row(child0=1, child1=-23550, child2=823668003, child3=-6580116972850632313, child4=3.816635312018178e+23, child5=8.600719030180411e+134, child6='\x0eM!R\x02ÇB\x1a¡^õ\x8cCÎµ\x1dpØX-Ùrz\x07\x9c¨\x1cÅ¨¡', child7=None, child8=datetime.date(5437, 1, 2), child9=None, child10=None)]), Row(a=7984374766242566542, sort_array(collect_set(b), true)=[Row(child0=-50, child1=-31138, child2=-1807864316, child3=-4041992126494898742, child4=4.065499056633676e+31, child5=7.554028173936504e-95, child6='ÆªOo©\x85·2Ôo!áP\x94\x9eþädÚ\x90õ4¢=L\x9däVi}', child7=True, child8=datetime.date(18, 7, 4), child9=datetime.datetime(6109, 8, 17, 22, 8, 3, 694767), child10=None), Row(child0=-27, child1=-24753, child2=-1726248225, child3=3307548583936611830, child4=-3.656386885506302e+18, child5=nan, child6='5\x92%w|ð\x9f\x98!\x887DÚdÌ$\x95Xò\xad\x98_ÿ>\x0b¡\tÇ(µ', child7=False, child8=datetime.date(9497, 10, 18), child9=datetime.datetime(8913, 8, 27, 22, 48, 29, 877849), child10=None), Row(child0=39, child1=-22177, child2=347245469, child3=5615010007311415123, child4=-0.16667649149894714, child5=-2.1002678349193272e+151, child6="%{³ÁaÃ\x13ò£|dÐ'Y\x04B½u\x99èÈ(!Ú\x13,ÜhÔþ", child7=False, child8=datetime.date(8508, 12, 19), child9=datetime.datetime(7713, 4, 6, 2, 18, 9, 495678), child10=None)]), Row(a=9223372036854775807, sort_array(collect_set(b), true)=[Row(child0=-50, child1=-31138, child2=-1807864316, child3=-4041992126494898742, child4=4.065499056633676e+31, child5=7.554028173936504e-95, child6='ÆªOo©\x85·2Ôo!áP\x94\x9eþädÚ\x90õ4¢=L\x9däVi}', child7=True, child8=datetime.date(18, 7, 4), child9=datetime.datetime(6109, 8, 17, 22, 8, 3, 694767), child10=None), Row(child0=-27, child1=-24753, child2=-1726248225, child3=3307548583936611830, child4=-3.656386885506302e+18, child5=nan, child6='5\x92%w|ð\x9f\x98!\x887DÚdÌ$\x95Xò\xad\x98_ÿ>\x0b¡\tÇ(µ', child7=False, child8=datetime.date(9497, 10, 18), child9=datetime.datetime(8913, 8, 27, 22, 48, 29, 877849), child10=None), Row(child0=39, child1=-22177, child2=347245469, child3=5615010007311415123, child4=-0.16667649149894714, child5=-2.1002678349193272e+151, child6="%{³ÁaÃ\x13ò£|dÐ'Y\x04B½u\x99èÈ(!Ú\x13,ÜhÔþ", child7=False, child8=datetime.date(8508, 12, 19), child9=datetime.datetime(7713, 4, 6, 2, 18, 9, 495678), child10=None)])]
GPU OUTPUT: [Row(a=-7540734677356764604, sort_array(collect_set(b), true)=[Row(child0=-110, child1=-11291, child2=358420087, child3=894377461322806935, child4=-1.6834864679694056e-31, child5=1.0303888663388996e-26, child6='¯óÏôÅkùê,dÌA~|×AÖ\x08VJé\x08yX^i{.\x1b\x83', child7=False, child8=datetime.date(4899, 3, 2), child9=datetime.datetime(6861, 9, 7, 19, 16, 50, 365146), child10=None), Row(child0=-61, child1=-18968, child2=639916889, child3=2251193521986836658, child4=1.7728260685861134e+25, child5=-1.6653918844116466e+240, child6=' 7jgv#\x1aÑÚªï\x8d\x9fð\x15\x83\x1fîìAæöl\x0bå\\\x90æ9â', child7=True, child8=datetime.date(353, 7, 23), child9=datetime.datetime(551, 6, 24, 14, 12, 58, 187381), child10=None), Row(child0=-41, child1=-24617, child2=848988934, child3=-7858370133784586516, child4=-1.740257146253725e-20, child5=3.5575370470599927e+284, child6='õ^\x1f·\x97ú·ÖF{/ZR*ø\x7fCýö\x05%A1Ð¶d\x04\x89øx', child7=True, child8=datetime.date(7802, 5, 13), child9=datetime.datetime(1279, 10, 18, 15, 49, 5, 515554), child10=None)]), Row(a=-5831592707909023540, sort_array(collect_set(b), true)=[Row(child0=-36, child1=28459, child2=2096953985, child3=1287002480498025462, child4=963.7640991210938, child5=9.376528689861087e+117, child6='\x05Â,ÀøYªe$«q;~\x04ëâ\x13h\x98ÇR\x04\x1e\x00©4@-\x0b°', child7=False, child8=datetime.date(8000, 2, 29), child9=datetime.datetime(5690, 4, 15, 13, 26, 32, 4190), child10=None), Row(child0=1, child1=-23550, child2=823668003, child3=-6580116972850632313, child4=3.816635312018178e+23, child5=8.600719030180411e+134, child6='\x0eM!R\x02ÇB\x1a¡^õ\x8cCÎµ\x1dpØX-Ùrz\x07\x9c¨\x1cÅ¨¡', child7=None, child8=datetime.date(5437, 1, 2), child9=None, child10=None)]), Row(a=-5133656973475552689, sort_array(collect_set(b), true)=[Row(child0=-108, child1=1267, child2=1917713880, child3=None, child4=None, child5=4.3214483342777574e-117, child6='[H\x89µ³hß\x14Æ\x12_XÕµoy\x08Y£àK;uk\x06\x15ç\x18\x02#', child7=True, child8=datetime.date(1590, 12, 27), child9=datetime.datetime(6166, 6, 5, 15, 21, 41, 824645), child10=None), Row(child0=-87, child1=18519, child2=105544272, child3=6446878210185090927, child4=None, child5=-4.2132496663975606e+247, child6='¦\r\x15\x16ep[|p\x9aË\x17ZZû\x82\x86\rê¼¨Ð³AúÍÿ\tÀò', child7=False, child8=datetime.date(3851, 2, 13), child9=datetime.datetime(297, 2, 6, 14, 13, 25, 35309), child10=None), Row(child0=6, child1=None, child2=-394822741, child3=-6874717541235784212, child4=4243308150784.0, child5=-3.638751059169508e+285, child6='M\x80²ÝÄ\x81¬m]ø\x14å\x05L·\x99®\x8e\x9a\x1a}oÝ\x01à\nãÔ\x84\x08', child7=False, child8=datetime.date(7026, 5, 5), child9=datetime.datetime(860, 3, 3, 14, 32, 46, 946330), child10=None)]), Row(a=-4426181692283497353, sort_array(collect_set(b), true)=[Row(child0=-50, child1=-31138, child2=-1807864316, child3=-4041992126494898742, child4=4.065499056633676e+31, child5=7.554028173936504e-95, child6='ÆªOo©\x85·2Ôo!áP\x94\x9eþädÚ\x90õ4¢=L\x9däVi}', child7=True, child8=datetime.date(18, 7, 4), child9=datetime.datetime(6109, 8, 17, 22, 8, 3, 694767), child10=None), Row(child0=-27, child1=-24753, child2=-1726248225, child3=3307548583936611830, child4=-3.656386885506302e+18, child5=nan, child6='5\x92%w|ð\x9f\x98!\x887DÚdÌ$\x95Xò\xad\x98_ÿ>\x0b¡\tÇ(µ', child7=False, child8=datetime.date(9497, 10, 18), child9=datetime.datetime(8913, 8, 27, 22, 48, 29, 877849), child10=None), Row(child0=-27, child1=-24753, child2=-1726248225, child3=3307548583936611830, child4=-3.656386885506302e+18, child5=nan, child6='5\x92%w|ð\x9f\x98!\x887DÚdÌ$\x95Xò\xad\x98_ÿ>\x0b¡\tÇ(µ', child7=False, child8=datetime.date(9497, 10, 18), child9=datetime.datetime(8913, 8, 27, 22, 48, 29, 877849), child10=None), Row(child0=39, child1=-22177, child2=347245469, child3=5615010007311415123, child4=-0.16667649149894714, child5=-2.1002678349193272e+151, child6="%{³ÁaÃ\x13ò£|dÐ'Y\x04B½u\x99èÈ(!Ú\x13,ÜhÔþ", child7=False, child8=datetime.date(8508, 12, 19), child9=datetime.datetime(7713, 4, 6, 2, 18, 9, 495678), child10=None)]), Row(a=-3917032101531217289, sort_array(collect_set(b), true)=[Row(child0=-108, child1=1267, child2=1917713880, child3=None, child4=None, child5=4.3214483342777574e-117, child6='[H\x89µ³hß\x14Æ\x12_XÕµoy\x08Y£àK;uk\x06\x15ç\x18\x02#', child7=True, child8=datetime.date(1590, 12, 27), child9=datetime.datetime(6166, 6, 5, 15, 21, 41, 824645), child10=None), Row(child0=-87, child1=18519, child2=105544272, child3=6446878210185090927, child4=None, child5=-4.2132496663975606e+247, child6='¦\r\x15\x16ep[|p\x9aË\x17ZZû\x82\x86\rê¼¨Ð³AúÍÿ\tÀò', child7=False, child8=datetime.date(3851, 2, 13), child9=datetime.datetime(297, 2, 6, 14, 13, 25, 35309), child10=None), Row(child0=6, child1=None, child2=-394822741, child3=-6874717541235784212, child4=4243308150784.0, child5=-3.638751059169508e+285, child6='M\x80²ÝÄ\x81¬m]ø\x14å\x05L·\x99®\x8e\x9a\x1a}oÝ\x01à\nãÔ\x84\x08', child7=False, child8=datetime.date(7026, 5, 5), child9=datetime.datetime(860, 3, 3, 14, 32, 46, 946330), child10=None)]), Row(a=-3502159106106506455, sort_array(collect_set(b), true)=[Row(child0=-108, child1=1267, child2=1917713880, child3=None, child4=None, child5=4.3214483342777574e-117, child6='[H\x89µ³hß\x14Æ\x12_XÕµoy\x08Y£àK;uk\x06\x15ç\x18\x02#', child7=True, child8=datetime.date(1590, 12, 27), child9=datetime.datetime(6166, 6, 5, 15, 21, 41, 824645), child10=None), Row(child0=-87, child1=18519, child2=105544272, child3=6446878210185090927, child4=None, child5=-4.2132496663975606e+247, child6='¦\r\x15\x16ep[|p\x9aË\x17ZZû\x82\x86\rê¼¨Ð³AúÍÿ\tÀò', child7=False, child8=datetime.date(3851, 2, 13), child9=datetime.datetime(297, 2, 6, 14, 13, 25, 35309), child10=None), Row(child0=6, child1=None, child2=-394822741, child3=-6874717541235784212, child4=4243308150784.0, child5=-3.638751059169508e+285, child6='M\x80²ÝÄ\x81¬m]ø\x14å\x05L·\x99®\x8e\x9a\x1a}oÝ\x01à\nãÔ\x84\x08', child7=False, child8=datetime.date(7026, 5, 5), child9=datetime.datetime(860, 3, 3, 14, 32, 46, 946330), child10=None)]), Row(a=-2697073954890740236, sort_array(collect_set(b), true)=[Row(child0=-110, child1=-11291, child2=358420087, child3=894377461322806935, child4=-1.6834864679694056e-31, child5=1.0303888663388996e-26, child6='¯óÏôÅkùê,dÌA~|×AÖ\x08VJé\x08yX^i{.\x1b\x83', child7=False, child8=datetime.date(4899, 3, 2), child9=datetime.datetime(6861, 9, 7, 19, 16, 50, 365146), child10=None), Row(child0=-61, child1=-18968, child2=639916889, child3=2251193521986836658, child4=1.7728260685861134e+25, child5=-1.6653918844116466e+240, child6=' 7jgv#\x1aÑÚªï\x8d\x9fð\x15\x83\x1fîìAæöl\x0bå\\\x90æ9â', child7=True, child8=datetime.date(353, 7, 23), child9=datetime.datetime(551, 6, 24, 14, 12, 58, 187381), child10=None), Row(child0=-41, child1=-24617, child2=848988934, child3=-7858370133784586516, child4=-1.740257146253725e-20, child5=3.5575370470599927e+284, child6='õ^\x1f·\x97ú·ÖF{/ZR*ø\x7fCýö\x05%A1Ð¶d\x04\x89øx', child7=True, child8=datetime.date(7802, 5, 13), child9=datetime.datetime(1279, 10, 18, 15, 49, 5, 515554), child10=None)]), Row(a=-2123199122092230623, sort_array(collect_set(b), true)=[Row(child0=-110, child1=-11291, child2=358420087, child3=894377461322806935, child4=-1.6834864679694056e-31, child5=1.0303888663388996e-26, child6='¯óÏôÅkùê,dÌA~|×AÖ\x08VJé\x08yX^i{.\x1b\x83', child7=False, child8=datetime.date(4899, 3, 2), child9=datetime.datetime(6861, 9, 7, 19, 16, 50, 365146), child10=None), Row(child0=-61, child1=-18968, child2=639916889, child3=2251193521986836658, child4=1.7728260685861134e+25, child5=-1.6653918844116466e+240, child6=' 7jgv#\x1aÑÚªï\x8d\x9fð\x15\x83\x1fîìAæöl\x0bå\\\x90æ9â', child7=True, child8=datetime.date(353, 7, 23), child9=datetime.datetime(551, 6, 24, 14, 12, 58, 187381), child10=None), Row(child0=-41, child1=-24617, child2=848988934, child3=-7858370133784586516, child4=-1.740257146253725e-20, child5=3.5575370470599927e+284, child6='õ^\x1f·\x97ú·ÖF{/ZR*ø\x7fCýö\x05%A1Ð¶d\x04\x89øx', child7=True, child8=datetime.date(7802, 5, 13), child9=datetime.datetime(1279, 10, 18, 15, 49, 5, 515554), child10=None)]), Row(a=-1, sort_array(collect_set(b), true)=[Row(child0=-36, child1=28459, child2=2096953985, child3=1287002480498025462, child4=963.7640991210938, child5=9.376528689861087e+117, child6='\x05Â,ÀøYªe$«q;~\x04ëâ\x13h\x98ÇR\x04\x1e\x00©4@-\x0b°', child7=False, child8=datetime.date(8000, 2, 29), child9=datetime.datetime(5690, 4, 15, 13, 26, 32, 4190), child10=None), Row(child0=1, child1=-23550, child2=823668003, child3=-6580116972850632313, child4=3.816635312018178e+23, child5=8.600719030180411e+134, child6='\x0eM!R\x02ÇB\x1a¡^õ\x8cCÎµ\x1dpØX-Ùrz\x07\x9c¨\x1cÅ¨¡', child7=None, child8=datetime.date(5437, 1, 2), child9=None, child10=None)]), Row(a=207981845540287738, sort_array(collect_set(b), true)=[Row(child0=-110, child1=-11291, child2=358420087, child3=894377461322806935, child4=-1.6834864679694056e-31, child5=1.0303888663388996e-26, child6='¯óÏôÅkùê,dÌA~|×AÖ\x08VJé\x08yX^i{.\x1b\x83', child7=False, child8=datetime.date(4899, 3, 2), child9=datetime.datetime(6861, 9, 7, 19, 16, 50, 365146), child10=None), Row(child0=-61, child1=-18968, child2=639916889, child3=2251193521986836658, child4=1.7728260685861134e+25, child5=-1.6653918844116466e+240, child6=' 7jgv#\x1aÑÚªï\x8d\x9fð\x15\x83\x1fîìAæöl\x0bå\\\x90æ9â', child7=True, child8=datetime.date(353, 7, 23), child9=datetime.datetime(551, 6, 24, 14, 12, 58, 187381), child10=None), Row(child0=-41, child1=-24617, child2=848988934, child3=-7858370133784586516, child4=-1.740257146253725e-20, child5=3.5575370470599927e+284, child6='õ^\x1f·\x97ú·ÖF{/ZR*ø\x7fCýö\x05%A1Ð¶d\x04\x89øx', child7=True, child8=datetime.date(7802, 5, 13), child9=datetime.datetime(1279, 10, 18, 15, 49, 5, 515554), child10=None)]), Row(a=393905103838704542, sort_array(collect_set(b), true)=[Row(child0=61, child1=-25185, child2=-2111287335, child3=-9223372036854775808, child4=None, child5=4.813757617752221e+57, child6=':\x9aLno»7þöµ\x00ý¦\x98vFWÊ\x9eeslr÷t±²ñ\x1eú', child7=True, child8=datetime.date(6996, 5, 13), child9=datetime.datetime(1691, 9, 13, 20, 49, 32, 239246), child10=None), Row(child0=84, child1=22049, child2=897674100, child3=7081246721334085708, child4=-5.445810266389943e+33, child5=-6.542279108216022e+206, child6='çìéô\x1cÀFS¥`\x04\x97ª¶\x14Ó\r¼\xa0¬ôÉe\x8dá~ì:£\x15', child7=False, child8=None, child9=datetime.datetime(2103, 9, 30, 11, 10, 53, 134710), child10=None), Row(child0=125, child1=9497, child2=2139679874, child3=7956328026836954420, child4=-243864059904.0, child5=2.1874890847202007e-206, child6='w\x96\x96\x1d\x8dÏ\x9b\x80\x86Úcy@5øäP\x86Ê\x81\x9coØr)\x8c{rY\x0c', child7=True, child8=datetime.date(3, 10, 7), child9=datetime.datetime(9259, 12, 8, 20, 39, 28, 751490), child10=None)]), Row(a=875130347651831881, sort_array(collect_set(b), true)=[Row(child0=61, child1=-25185, child2=-2111287335, child3=-9223372036854775808, child4=None, child5=4.813757617752221e+57, child6=':\x9aLno»7þöµ\x00ý¦\x98vFWÊ\x9eeslr÷t±²ñ\x1eú', child7=True, child8=datetime.date(6996, 5, 13), child9=datetime.datetime(1691, 9, 13, 20, 49, 32, 239246), child10=None), Row(child0=84, child1=22049, child2=897674100, child3=7081246721334085708, child4=-5.445810266389943e+33, child5=-6.542279108216022e+206, child6='çìéô\x1cÀFS¥`\x04\x97ª¶\x14Ó\r¼\xa0¬ôÉe\x8dá~ì:£\x15', child7=False, child8=None, child9=datetime.datetime(2103, 9, 30, 11, 10, 53, 134710), child10=None), Row(child0=125, child1=9497, child2=2139679874, child3=7956328026836954420, child4=-243864059904.0, child5=2.1874890847202007e-206, child6='w\x96\x96\x1d\x8dÏ\x9b\x80\x86Úcy@5øäP\x86Ê\x81\x9coØr)\x8c{rY\x0c', child7=True, child8=datetime.date(3, 10, 7), child9=datetime.datetime(9259, 12, 8, 20, 39, 28, 751490), child10=None)]), Row(a=4751953708995107450, sort_array(collect_set(b), true)=[Row(child0=61, child1=-25185, child2=-2111287335, child3=-9223372036854775808, child4=None, child5=4.813757617752221e+57, child6=':\x9aLno»7þöµ\x00ý¦\x98vFWÊ\x9eeslr÷t±²ñ\x1eú', child7=True, child8=datetime.date(6996, 5, 13), child9=datetime.datetime(1691, 9, 13, 20, 49, 32, 239246), child10=None), Row(child0=84, child1=22049, child2=897674100, child3=7081246721334085708, child4=-5.445810266389943e+33, child5=-6.542279108216022e+206, child6='çìéô\x1cÀFS¥`\x04\x97ª¶\x14Ó\r¼\xa0¬ôÉe\x8dá~ì:£\x15', child7=False, child8=None, child9=datetime.datetime(2103, 9, 30, 11, 10, 53, 134710), child10=None), Row(child0=125, child1=9497, child2=2139679874, child3=7956328026836954420, child4=-243864059904.0, child5=2.1874890847202007e-206, child6='w\x96\x96\x1d\x8dÏ\x9b\x80\x86Úcy@5øäP\x86Ê\x81\x9coØr)\x8c{rY\x0c', child7=True, child8=datetime.date(3, 10, 7), child9=datetime.datetime(9259, 12, 8, 20, 39, 28, 751490), child10=None)]), Row(a=6084712057446794809, sort_array(collect_set(b), true)=[Row(child0=-36, child1=28459, child2=2096953985, child3=1287002480498025462, child4=963.7640991210938, child5=9.376528689861087e+117, child6='\x05Â,ÀøYªe$«q;~\x04ëâ\x13h\x98ÇR\x04\x1e\x00©4@-\x0b°', child7=False, child8=datetime.date(8000, 2, 29), child9=datetime.datetime(5690, 4, 15, 13, 26, 32, 4190), child10=None), Row(child0=1, child1=-23550, child2=823668003, child3=-6580116972850632313, child4=3.816635312018178e+23, child5=8.600719030180411e+134, child6='\x0eM!R\x02ÇB\x1a¡^õ\x8cCÎµ\x1dpØX-Ùrz\x07\x9c¨\x1cÅ¨¡', child7=None, child8=datetime.date(5437, 1, 2), child9=None, child10=None)]), Row(a=7198729688045931692, sort_array(collect_set(b), true)=[Row(child0=-108, child1=1267, child2=1917713880, child3=None, child4=None, child5=4.3214483342777574e-117, child6='[H\x89µ³hß\x14Æ\x12_XÕµoy\x08Y£àK;uk\x06\x15ç\x18\x02#', child7=True, child8=datetime.date(1590, 12, 27), child9=datetime.datetime(6166, 6, 5, 15, 21, 41, 824645), child10=None), Row(child0=-87, child1=18519, child2=105544272, child3=6446878210185090927, child4=None, child5=-4.2132496663975606e+247, child6='¦\r\x15\x16ep[|p\x9aË\x17ZZû\x82\x86\rê¼¨Ð³AúÍÿ\tÀò', child7=False, child8=datetime.date(3851, 2, 13), child9=datetime.datetime(297, 2, 6, 14, 13, 25, 35309), child10=None), Row(child0=6, child1=None, child2=-394822741, child3=-6874717541235784212, child4=4243308150784.0, child5=-3.638751059169508e+285, child6='M\x80²ÝÄ\x81¬m]ø\x14å\x05L·\x99®\x8e\x9a\x1a}oÝ\x01à\nãÔ\x84\x08', child7=False, child8=datetime.date(7026, 5, 5), child9=datetime.datetime(860, 3, 3, 14, 32, 46, 946330), child10=None)]), Row(a=7528354001793048440, sort_array(collect_set(b), true)=[Row(child0=61, child1=-25185, child2=-2111287335, child3=-9223372036854775808, child4=None, child5=4.813757617752221e+57, child6=':\x9aLno»7þöµ\x00ý¦\x98vFWÊ\x9eeslr÷t±²ñ\x1eú', child7=True, child8=datetime.date(6996, 5, 13), child9=datetime.datetime(1691, 9, 13, 20, 49, 32, 239246), child10=None), Row(child0=84, child1=22049, child2=897674100, child3=7081246721334085708, child4=-5.445810266389943e+33, child5=-6.542279108216022e+206, child6='çìéô\x1cÀFS¥`\x04\x97ª¶\x14Ó\r¼\xa0¬ôÉe\x8dá~ì:£\x15', child7=False, child8=None, child9=datetime.datetime(2103, 9, 30, 11, 10, 53, 134710), child10=None), Row(child0=125, child1=9497, child2=2139679874, child3=7956328026836954420, child4=-243864059904.0, child5=2.1874890847202007e-206, child6='w\x96\x96\x1d\x8dÏ\x9b\x80\x86Úcy@5øäP\x86Ê\x81\x9coØr)\x8c{rY\x0c', child7=True, child8=datetime.date(3, 10, 7), child9=datetime.datetime(9259, 12, 8, 20, 39, 28, 751490), child10=None)]), Row(a=7618709293599214015, sort_array(collect_set(b), true)=[Row(child0=-36, child1=28459, child2=2096953985, child3=1287002480498025462, child4=963.7640991210938, child5=9.376528689861087e+117, child6='\x05Â,ÀøYªe$«q;~\x04ëâ\x13h\x98ÇR\x04\x1e\x00©4@-\x0b°', child7=False, child8=datetime.date(8000, 2, 29), child9=datetime.datetime(5690, 4, 15, 13, 26, 32, 4190), child10=None), Row(child0=1, child1=-23550, child2=823668003, child3=-6580116972850632313, child4=3.816635312018178e+23, child5=8.600719030180411e+134, child6='\x0eM!R\x02ÇB\x1a¡^õ\x8cCÎµ\x1dpØX-Ùrz\x07\x9c¨\x1cÅ¨¡', child7=None, child8=datetime.date(5437, 1, 2), child9=None, child10=None)]), Row(a=7984374766242566542, sort_array(collect_set(b), true)=[Row(child0=-50, child1=-31138, child2=-1807864316, child3=-4041992126494898742, child4=4.065499056633676e+31, child5=7.554028173936504e-95, child6='ÆªOo©\x85·2Ôo!áP\x94\x9eþädÚ\x90õ4¢=L\x9däVi}', child7=True, child8=datetime.date(18, 7, 4), child9=datetime.datetime(6109, 8, 17, 22, 8, 3, 694767), child10=None), Row(child0=-27, child1=-24753, child2=-1726248225, child3=3307548583936611830, child4=-3.656386885506302e+18, child5=nan, child6='5\x92%w|ð\x9f\x98!\x887DÚdÌ$\x95Xò\xad\x98_ÿ>\x0b¡\tÇ(µ', child7=False, child8=datetime.date(9497, 10, 18), child9=datetime.datetime(8913, 8, 27, 22, 48, 29, 877849), child10=None), Row(child0=39, child1=-22177, child2=347245469, child3=5615010007311415123, child4=-0.16667649149894714, child5=-2.1002678349193272e+151, child6="%{³ÁaÃ\x13ò£|dÐ'Y\x04B½u\x99èÈ(!Ú\x13,ÜhÔþ", child7=False, child8=datetime.date(8508, 12, 19), child9=datetime.datetime(7713, 4, 6, 2, 18, 9, 495678), child10=None)]), Row(a=9223372036854775807, sort_array(collect_set(b), true)=[Row(child0=-50, child1=-31138, child2=-1807864316, child3=-4041992126494898742, child4=4.065499056633676e+31, child5=7.554028173936504e-95, child6='ÆªOo©\x85·2Ôo!áP\x94\x9eþädÚ\x90õ4¢=L\x9däVi}', child7=True, child8=datetime.date(18, 7, 4), child9=datetime.datetime(6109, 8, 17, 22, 8, 3, 694767), child10=None), Row(child0=-27, child1=-24753, child2=-1726248225, child3=3307548583936611830, child4=-3.656386885506302e+18, child5=nan, child6='5\x92%w|ð\x9f\x98!\x887DÚdÌ$\x95Xò\xad\x98_ÿ>\x0b¡\tÇ(µ', child7=False, child8=datetime.date(9497, 10, 18), child9=datetime.datetime(8913, 8, 27, 22, 48, 29, 877849), child10=None), Row(child0=-27, child1=-24753, child2=-1726248225, child3=3307548583936611830, child4=-3.656386885506302e+18, child5=nan, child6='5\x92%w|ð\x9f\x98!\x887DÚdÌ$\x95Xò\xad\x98_ÿ>\x0b¡\tÇ(µ', child7=False, child8=datetime.date(9497, 10, 18), child9=datetime.datetime(8913, 8, 27, 22, 48, 29, 877849), child10=None), Row(child0=-27, child1=-24753, child2=-1726248225, child3=3307548583936611830, child4=-3.656386885506302e+18, child5=nan, child6='5\x92%w|ð\x9f\x98!\x887DÚdÌ$\x95Xò\xad\x98_ÿ>\x0b¡\tÇ(µ', child7=False, child8=datetime.date(9497, 10, 18), child9=datetime.datetime(8913, 8, 27, 22, 48, 29, 877849), child10=None), Row(child0=-27, child1=-24753, child2=-1726248225, child3=3307548583936611830, child4=-3.656386885506302e+18, child5=nan, child6='5\x92%w|ð\x9f\x98!\x887DÚdÌ$\x95Xò\xad\x98_ÿ>\x0b¡\tÇ(µ', child7=False, child8=datetime.date(9497, 10, 18), child9=datetime.datetime(8913, 8, 27, 22, 48, 29, 877849), child10=None), Row(child0=39, child1=-22177, child2=347245469, child3=5615010007311415123, child4=-0.16667649149894714, child5=-2.1002678349193272e+151, child6="%{³ÁaÃ\x13ò£|dÐ'Y\x04B½u\x99èÈ(!Ú\x13,ÜhÔþ", child7=False, child8=datetime.date(8508, 12, 19), child9=datetime.datetime(7713, 4, 6, 2, 18, 9, 495678), child10=None)])]
_ test_hash_reduction_collect_set_on_nested_type[[('a', RepeatSeq(Long)), ('b', RepeatSeq(Struct(['child0', Byte],['child1', Short],['child2', Integer],['child3', Long],['child4', Float],['child5', Double],['child6', String],['child7', Boolean],['child8', Date],['child9', Timestamp],['child10', Null])))]] _
[gw2] linux -- Python 3.8.3 /home/haoyangl/.pyenv/versions/3.8.3/bin/python

data_gen = [('a', RepeatSeq(Long)), ('b', RepeatSeq(Struct(['child0', Byte],['child1', Short],['child2', Integer],['child3', Long...,['child5', Double],['child6', String],['child7', Boolean],['child8', Date],['child9', Timestamp],['child10', Null])))]

    @ignore_order(local=True)
    @pytest.mark.parametrize('data_gen', _gen_data_for_collect_set_op, ids=idfn)
    # @pytest.mark.xfail(condition=is_before_spark_330(), reason='https://github.com/NVIDIA/spark-rapids/issues/8716')
    def test_hash_reduction_collect_set_on_nested_type(data_gen):
>       assert_gpu_and_cpu_are_equal_collect(
            lambda spark: gen_df(spark, data_gen, length=100)
                .agg(f.sort_array(f.collect_set('b'))))

../../src/main/python/hash_aggregate_test.py:736:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../src/main/python/asserts.py:561: in assert_gpu_and_cpu_are_equal_collect
    _assert_gpu_and_cpu_are_equal(func, 'COLLECT', conf=conf, is_cpu_first=is_cpu_first)
../../src/main/python/asserts.py:492: in _assert_gpu_and_cpu_are_equal
    assert_equal(from_cpu, from_gpu)
../../src/main/python/asserts.py:105: in assert_equal
    _assert_equal(cpu, gpu, float_check=get_float_check(), path=[])
../../src/main/python/asserts.py:41: in _assert_equal
    _assert_equal(cpu[index], gpu[index], float_check, path + [index])
../../src/main/python/asserts.py:34: in _assert_equal
    _assert_equal(cpu[field], gpu[field], float_check, path + [field])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cpu = [Row(child0=-110, child1=-11291, child2=358420087, child3=894377461322806935, child4=-1.6834864679694056e-31, child5=1...True, child8=datetime.date(7802, 5, 13), child9=datetime.datetime(1279, 10, 18, 15, 49, 5, 515554), child10=None), ...]
gpu = [Row(child0=-110, child1=-11291, child2=358420087, child3=894377461322806935, child4=-1.6834864679694056e-31, child5=1...True, child8=datetime.date(7802, 5, 13), child9=datetime.datetime(1279, 10, 18, 15, 49, 5, 515554), child10=None), ...]
float_check = <function get_float_check.<locals>.<lambda> at 0x7f617169f3a0>, path = [0, 'sort_array(collect_set(b), true)']

    def _assert_equal(cpu, gpu, float_check, path):
        t = type(cpu)
        if (t is Row):
            assert len(cpu) == len(gpu), "CPU and GPU row have different lengths at {} CPU: {} GPU: {}".format(path, len(cpu), len(gpu))
            if hasattr(cpu, "__fields__") and hasattr(gpu, "__fields__"):
                assert cpu.__fields__ == gpu.__fields__, "CPU and GPU row have different fields at {} CPU: {} GPU: {}".format(path, cpu.__fields__, gpu.__fields__)
                for field in cpu.__fields__:
                    _assert_equal(cpu[field], gpu[field], float_check, path + [field])
            else:
                for index in range(len(cpu)):
                    _assert_equal(cpu[index], gpu[index], float_check, path + [index])
        elif (t is list):
>           assert len(cpu) == len(gpu), "CPU and GPU list have different lengths at {} CPU: {} GPU: {}".format(path, len(cpu), len(gpu))
E           AssertionError: CPU and GPU list have different lengths at [0, 'sort_array(collect_set(b), true)'] CPU: 14 GPU: 20

../../src/main/python/asserts.py:39: AssertionError
----------------------------------------------------------- Captured stdout call ------------------------------------------------------------
### CPU RUN ###
### GPU RUN ###
### COLLECT: GPU TOOK 1.2127246856689453 CPU TOOK 4.060694456100464 ###
CPU OUTPUT: [Row(sort_array(collect_set(b), true)=[Row(child0=-110, child1=-11291, child2=358420087, child3=894377461322806935, child4=-1.6834864679694056e-31, child5=1.0303888663388996e-26, child6='¯óÏôÅkùê,dÌA~|×AÖ\x08VJé\x08yX^i{.\x1b\x83', child7=False, child8=datetime.date(4899, 3, 2), child9=datetime.datetime(6861, 9, 7, 19, 16, 50, 365146), child10=None), Row(child0=-108, child1=1267, child2=1917713880, child3=None, child4=None, child5=4.3214483342777574e-117, child6='[H\x89µ³hß\x14Æ\x12_XÕµoy\x08Y£àK;uk\x06\x15ç\x18\x02#', child7=True, child8=datetime.date(1590, 12, 27), child9=datetime.datetime(6166, 6, 5, 15, 21, 41, 824645), child10=None), Row(child0=-87, child1=18519, child2=105544272, child3=6446878210185090927, child4=None, child5=-4.2132496663975606e+247, child6='¦\r\x15\x16ep[|p\x9aË\x17ZZû\x82\x86\rê¼¨Ð³AúÍÿ\tÀò', child7=False, child8=datetime.date(3851, 2, 13), child9=datetime.datetime(297, 2, 6, 14, 13, 25, 35309), child10=None), Row(child0=-61, child1=-18968, child2=639916889, child3=2251193521986836658, child4=1.7728260685861134e+25, child5=-1.6653918844116466e+240, child6=' 7jgv#\x1aÑÚªï\x8d\x9fð\x15\x83\x1fîìAæöl\x0bå\\\x90æ9â', child7=True, child8=datetime.date(353, 7, 23), child9=datetime.datetime(551, 6, 24, 14, 12, 58, 187381), child10=None), Row(child0=-50, child1=-31138, child2=-1807864316, child3=-4041992126494898742, child4=4.065499056633676e+31, child5=7.554028173936504e-95, child6='ÆªOo©\x85·2Ôo!áP\x94\x9eþädÚ\x90õ4¢=L\x9däVi}', child7=True, child8=datetime.date(18, 7, 4), child9=datetime.datetime(6109, 8, 17, 22, 8, 3, 694767), child10=None), Row(child0=-41, child1=-24617, child2=848988934, child3=-7858370133784586516, child4=-1.740257146253725e-20, child5=3.5575370470599927e+284, child6='õ^\x1f·\x97ú·ÖF{/ZR*ø\x7fCýö\x05%A1Ð¶d\x04\x89øx', child7=True, child8=datetime.date(7802, 5, 13), child9=datetime.datetime(1279, 10, 18, 15, 49, 5, 515554), child10=None), Row(child0=-36, child1=28459, child2=2096953985, child3=1287002480498025462, child4=963.7640991210938, child5=9.376528689861087e+117, child6='\x05Â,ÀøYªe$«q;~\x04ëâ\x13h\x98ÇR\x04\x1e\x00©4@-\x0b°', child7=False, child8=datetime.date(8000, 2, 29), child9=datetime.datetime(5690, 4, 15, 13, 26, 32, 4190), child10=None), Row(child0=-27, child1=-24753, child2=-1726248225, child3=3307548583936611830, child4=-3.656386885506302e+18, child5=nan, child6='5\x92%w|ð\x9f\x98!\x887DÚdÌ$\x95Xò\xad\x98_ÿ>\x0b¡\tÇ(µ', child7=False, child8=datetime.date(9497, 10, 18), child9=datetime.datetime(8913, 8, 27, 22, 48, 29, 877849), child10=None), Row(child0=1, child1=-23550, child2=823668003, child3=-6580116972850632313, child4=3.816635312018178e+23, child5=8.600719030180411e+134, child6='\x0eM!R\x02ÇB\x1a¡^õ\x8cCÎµ\x1dpØX-Ùrz\x07\x9c¨\x1cÅ¨¡', child7=None, child8=datetime.date(5437, 1, 2), child9=None, child10=None), Row(child0=6, child1=None, child2=-394822741, child3=-6874717541235784212, child4=4243308150784.0, child5=-3.638751059169508e+285, child6='M\x80²ÝÄ\x81¬m]ø\x14å\x05L·\x99®\x8e\x9a\x1a}oÝ\x01à\nãÔ\x84\x08', child7=False, child8=datetime.date(7026, 5, 5), child9=datetime.datetime(860, 3, 3, 14, 32, 46, 946330), child10=None), Row(child0=39, child1=-22177, child2=347245469, child3=5615010007311415123, child4=-0.16667649149894714, child5=-2.1002678349193272e+151, child6="%{³ÁaÃ\x13ò£|dÐ'Y\x04B½u\x99èÈ(!Ú\x13,ÜhÔþ", child7=False, child8=datetime.date(8508, 12, 19), child9=datetime.datetime(7713, 4, 6, 2, 18, 9, 495678), child10=None), Row(child0=61, child1=-25185, child2=-2111287335, child3=-9223372036854775808, child4=None, child5=4.813757617752221e+57, child6=':\x9aLno»7þöµ\x00ý¦\x98vFWÊ\x9eeslr÷t±²ñ\x1eú', child7=True, child8=datetime.date(6996, 5, 13), child9=datetime.datetime(1691, 9, 13, 20, 49, 32, 239246), child10=None), Row(child0=84, child1=22049, child2=897674100, child3=7081246721334085708, child4=-5.445810266389943e+33, child5=-6.542279108216022e+206, child6='çìéô\x1cÀFS¥`\x04\x97ª¶\x14Ó\r¼\xa0¬ôÉe\x8dá~ì:£\x15', child7=False, child8=None, child9=datetime.datetime(2103, 9, 30, 11, 10, 53, 134710), child10=None), Row(child0=125, child1=9497, child2=2139679874, child3=7956328026836954420, child4=-243864059904.0, child5=2.1874890847202007e-206, child6='w\x96\x96\x1d\x8dÏ\x9b\x80\x86Úcy@5øäP\x86Ê\x81\x9coØr)\x8c{rY\x0c', child7=True, child8=datetime.date(3, 10, 7), child9=datetime.datetime(9259, 12, 8, 20, 39, 28, 751490), child10=None)])]
GPU OUTPUT: [Row(sort_array(collect_set(b), true)=[Row(child0=-110, child1=-11291, child2=358420087, child3=894377461322806935, child4=-1.6834864679694056e-31, child5=1.0303888663388996e-26, child6='¯óÏôÅkùê,dÌA~|×AÖ\x08VJé\x08yX^i{.\x1b\x83', child7=False, child8=datetime.date(4899, 3, 2), child9=datetime.datetime(6861, 9, 7, 19, 16, 50, 365146), child10=None), Row(child0=-108, child1=1267, child2=1917713880, child3=None, child4=None, child5=4.3214483342777574e-117, child6='[H\x89µ³hß\x14Æ\x12_XÕµoy\x08Y£àK;uk\x06\x15ç\x18\x02#', child7=True, child8=datetime.date(1590, 12, 27), child9=datetime.datetime(6166, 6, 5, 15, 21, 41, 824645), child10=None), Row(child0=-87, child1=18519, child2=105544272, child3=6446878210185090927, child4=None, child5=-4.2132496663975606e+247, child6='¦\r\x15\x16ep[|p\x9aË\x17ZZû\x82\x86\rê¼¨Ð³AúÍÿ\tÀò', child7=False, child8=datetime.date(3851, 2, 13), child9=datetime.datetime(297, 2, 6, 14, 13, 25, 35309), child10=None), Row(child0=-61, child1=-18968, child2=639916889, child3=2251193521986836658, child4=1.7728260685861134e+25, child5=-1.6653918844116466e+240, child6=' 7jgv#\x1aÑÚªï\x8d\x9fð\x15\x83\x1fîìAæöl\x0bå\\\x90æ9â', child7=True, child8=datetime.date(353, 7, 23), child9=datetime.datetime(551, 6, 24, 14, 12, 58, 187381), child10=None), Row(child0=-50, child1=-31138, child2=-1807864316, child3=-4041992126494898742, child4=4.065499056633676e+31, child5=7.554028173936504e-95, child6='ÆªOo©\x85·2Ôo!áP\x94\x9eþädÚ\x90õ4¢=L\x9däVi}', child7=True, child8=datetime.date(18, 7, 4), child9=datetime.datetime(6109, 8, 17, 22, 8, 3, 694767), child10=None), Row(child0=-41, child1=-24617, child2=848988934, child3=-7858370133784586516, child4=-1.740257146253725e-20, child5=3.5575370470599927e+284, child6='õ^\x1f·\x97ú·ÖF{/ZR*ø\x7fCýö\x05%A1Ð¶d\x04\x89øx', child7=True, child8=datetime.date(7802, 5, 13), child9=datetime.datetime(1279, 10, 18, 15, 49, 5, 515554), child10=None), Row(child0=-36, child1=28459, child2=2096953985, child3=1287002480498025462, child4=963.7640991210938, child5=9.376528689861087e+117, child6='\x05Â,ÀøYªe$«q;~\x04ëâ\x13h\x98ÇR\x04\x1e\x00©4@-\x0b°', child7=False, child8=datetime.date(8000, 2, 29), child9=datetime.datetime(5690, 4, 15, 13, 26, 32, 4190), child10=None), Row(child0=-27, child1=-24753, child2=-1726248225, child3=3307548583936611830, child4=-3.656386885506302e+18, child5=nan, child6='5\x92%w|ð\x9f\x98!\x887DÚdÌ$\x95Xò\xad\x98_ÿ>\x0b¡\tÇ(µ', child7=False, child8=datetime.date(9497, 10, 18), child9=datetime.datetime(8913, 8, 27, 22, 48, 29, 877849), child10=None), Row(child0=-27, child1=-24753, child2=-1726248225, child3=3307548583936611830, child4=-3.656386885506302e+18, child5=nan, child6='5\x92%w|ð\x9f\x98!\x887DÚdÌ$\x95Xò\xad\x98_ÿ>\x0b¡\tÇ(µ', child7=False, child8=datetime.date(9497, 10, 18), child9=datetime.datetime(8913, 8, 27, 22, 48, 29, 877849), child10=None), Row(child0=-27, child1=-24753, child2=-1726248225, child3=3307548583936611830, child4=-3.656386885506302e+18, child5=nan, child6='5\x92%w|ð\x9f\x98!\x887DÚdÌ$\x95Xò\xad\x98_ÿ>\x0b¡\tÇ(µ', child7=False, child8=datetime.date(9497, 10, 18), child9=datetime.datetime(8913, 8, 27, 22, 48, 29, 877849), child10=None), Row(child0=-27, child1=-24753, child2=-1726248225, child3=3307548583936611830, child4=-3.656386885506302e+18, child5=nan, child6='5\x92%w|ð\x9f\x98!\x887DÚdÌ$\x95Xò\xad\x98_ÿ>\x0b¡\tÇ(µ', child7=False, child8=datetime.date(9497, 10, 18), child9=datetime.datetime(8913, 8, 27, 22, 48, 29, 877849), child10=None), Row(child0=-27, child1=-24753, child2=-1726248225, child3=3307548583936611830, child4=-3.656386885506302e+18, child5=nan, child6='5\x92%w|ð\x9f\x98!\x887DÚdÌ$\x95Xò\xad\x98_ÿ>\x0b¡\tÇ(µ', child7=False, child8=datetime.date(9497, 10, 18), child9=datetime.datetime(8913, 8, 27, 22, 48, 29, 877849), child10=None), Row(child0=-27, child1=-24753, child2=-1726248225, child3=3307548583936611830, child4=-3.656386885506302e+18, child5=nan, child6='5\x92%w|ð\x9f\x98!\x887DÚdÌ$\x95Xò\xad\x98_ÿ>\x0b¡\tÇ(µ', child7=False, child8=datetime.date(9497, 10, 18), child9=datetime.datetime(8913, 8, 27, 22, 48, 29, 877849), child10=None), Row(child0=-27, child1=-24753, child2=-1726248225, child3=3307548583936611830, child4=-3.656386885506302e+18, child5=nan, child6='5\x92%w|ð\x9f\x98!\x887DÚdÌ$\x95Xò\xad\x98_ÿ>\x0b¡\tÇ(µ', child7=False, child8=datetime.date(9497, 10, 18), child9=datetime.datetime(8913, 8, 27, 22, 48, 29, 877849), child10=None), Row(child0=1, child1=-23550, child2=823668003, child3=-6580116972850632313, child4=3.816635312018178e+23, child5=8.600719030180411e+134, child6='\x0eM!R\x02ÇB\x1a¡^õ\x8cCÎµ\x1dpØX-Ùrz\x07\x9c¨\x1cÅ¨¡', child7=None, child8=datetime.date(5437, 1, 2), child9=None, child10=None), Row(child0=6, child1=None, child2=-394822741, child3=-6874717541235784212, child4=4243308150784.0, child5=-3.638751059169508e+285, child6='M\x80²ÝÄ\x81¬m]ø\x14å\x05L·\x99®\x8e\x9a\x1a}oÝ\x01à\nãÔ\x84\x08', child7=False, child8=datetime.date(7026, 5, 5), child9=datetime.datetime(860, 3, 3, 14, 32, 46, 946330), child10=None), Row(child0=39, child1=-22177, child2=347245469, child3=5615010007311415123, child4=-0.16667649149894714, child5=-2.1002678349193272e+151, child6="%{³ÁaÃ\x13ò£|dÐ'Y\x04B½u\x99èÈ(!Ú\x13,ÜhÔþ", child7=False, child8=datetime.date(8508, 12, 19), child9=datetime.datetime(7713, 4, 6, 2, 18, 9, 495678), child10=None), Row(child0=61, child1=-25185, child2=-2111287335, child3=-9223372036854775808, child4=None, child5=4.813757617752221e+57, child6=':\x9aLno»7þöµ\x00ý¦\x98vFWÊ\x9eeslr÷t±²ñ\x1eú', child7=True, child8=datetime.date(6996, 5, 13), child9=datetime.datetime(1691, 9, 13, 20, 49, 32, 239246), child10=None), Row(child0=84, child1=22049, child2=897674100, child3=7081246721334085708, child4=-5.445810266389943e+33, child5=-6.542279108216022e+206, child6='çìéô\x1cÀFS¥`\x04\x97ª¶\x14Ó\r¼\xa0¬ôÉe\x8dá~ì:£\x15', child7=False, child8=None, child9=datetime.datetime(2103, 9, 30, 11, 10, 53, 134710), child10=None), Row(child0=125, child1=9497, child2=2139679874, child3=7956328026836954420, child4=-243864059904.0, child5=2.1874890847202007e-206, child6='w\x96\x96\x1d\x8dÏ\x9b\x80\x86Úcy@5øäP\x86Ê\x81\x9coØr)\x8c{rY\x0c', child7=True, child8=datetime.date(3, 10, 7), child9=datetime.datetime(9259, 12, 8, 20, 39, 28, 751490), child10=None)])]
- generated xml file: /home/haoyangl/spark-rapids/integration_tests/target/run_dir-20230717100401-K55a/TEST-pytest-1689559441610286928.xml --
========================================================== short test summary info ==========================================================
FAILED ../../src/main/python/hash_aggregate_test.py::test_hash_groupby_collect_set_on_nested_type[[('a', RepeatSeq(Long)), ('b', RepeatSeq(Struct(['child0', Byte],['child1', Short],['child2', Integer],['child3', Long],['child4', Float],['child5', Double],['child6', String],['child7', Boolean],['child8', Date],['child9', Timestamp],['child10', Null])))]][INJECT_OOM, IGNORE_ORDER({'local': True})]
FAILED ../../src/main/python/hash_aggregate_test.py::test_hash_reduction_collect_set_on_nested_type[[('a', RepeatSeq(Long)), ('b', RepeatSeq(Struct(['child0', Byte],['child1', Short],['child2', Integer],['child3', Long],['child4', Float],['child5', Double],['child6', String],['child7', Boolean],['child8', Date],['child9', Timestamp],['child10', Null])))]][INJECT_OOM, IGNORE_ORDER({'local': True})]
```

**Expected behavior**
These cases should pass.

**Environment details (please complete the following information)**
Latest code with spark 311. (I also tested them with spark 330 and they can pass).

another failure got seen in internal CI `JDK17-nightly` ID: 62 (spark 330) looks like this is an intermittent error

```
[2023-07-18T05:16:06.967Z] FAILED ../../src/main/python/hash_aggregate_test.py::test_hash_groupby_collect_set_on_nested_type[[('a', RepeatSeq(Long)), ('b', RepeatSeq(Struct(['child0', Byte],['child1', Short],['child2', Integer],['child3', Long],['child4', Float],['child5', Double],['child6', String],['child7', Boolean],['child8', Date],['child9', Timestamp],['child10', Null])))]][INJECT_OOM, IGNORE_ORDER({'local': True})] - AssertionError: CPU and GPU list have different lengths at [3, 'sort_array(collect_set(b), true)'] CPU: 3 GPU: 4

[2023-07-18T05:16:06.967Z] FAILED ../../src/main/python/hash_aggregate_test.py::test_hash_reduction_collect_set_on_nested_type[[('a', RepeatSeq(Long)), ('b', RepeatSeq(Struct(['child0', Byte],['child1', Short],['child2', Integer],['child3', Long],['child4', Float],['child5', Double],['child6', String],['child7', Boolean],['child8', Date],['child9', Timestamp],['child10', Null])))]][IGNORE_ORDER({'local': True})] - AssertionError: CPU and GPU list have different lengths at [0, 'sort_array(collect_set(b), true)'] CPU: 14 GPU: 20
```

mismatched output,
```
[2023-07-18T05:16:06.966Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2023-07-18T05:16:06.966Z] 

[2023-07-18T05:16:06.966Z] cpu = [Row(child0=-110, child1=-11291, child2=358420087, child3=894377461322806935, child4=-1.6834864679694056e-31, child5=1...True, child8=datetime.date(7802, 5, 13), child9=datetime.datetime(1279, 10, 18, 15, 49, 5, 515554), child10=None), ...]

[2023-07-18T05:16:06.966Z] gpu = [Row(child0=-110, child1=-11291, child2=358420087, child3=894377461322806935, child4=-1.6834864679694056e-31, child5=1...True, child8=datetime.date(7802, 5, 13), child9=datetime.datetime(1279, 10, 18, 15, 49, 5, 515554), child10=None), ...]

[2023-07-18T05:16:06.966Z] float_check = <function get_float_check.<locals>.<lambda> at 0x7f72f587bca0>

[2023-07-18T05:16:06.966Z] path = [0, 'sort_array(collect_set(b), true)']

[2023-07-18T05:16:06.966Z] 

[2023-07-18T05:16:06.966Z]     def _assert_equal(cpu, gpu, float_check, path):

[2023-07-18T05:16:06.966Z]         t = type(cpu)

[2023-07-18T05:16:06.966Z]         if (t is Row):

[2023-07-18T05:16:06.966Z]             assert len(cpu) == len(gpu), "CPU and GPU row have different lengths at {} CPU: {} GPU: {}".format(path, len(cpu), len(gpu))

[2023-07-18T05:16:06.966Z]             if hasattr(cpu, "__fields__") and hasattr(gpu, "__fields__"):

[2023-07-18T05:16:06.966Z]                 assert cpu.__fields__ == gpu.__fields__, "CPU and GPU row have different fields at {} CPU: {} GPU: {}".format(path, cpu.__fields__, gpu.__fields__)

[2023-07-18T05:16:06.966Z]                 for field in cpu.__fields__:

[2023-07-18T05:16:06.966Z]                     _assert_equal(cpu[field], gpu[field], float_check, path + [field])

[2023-07-18T05:16:06.966Z]             else:

[2023-07-18T05:16:06.966Z]                 for index in range(len(cpu)):

[2023-07-18T05:16:06.966Z]                     _assert_equal(cpu[index], gpu[index], float_check, path + [index])

[2023-07-18T05:16:06.966Z]         elif (t is list):

[2023-07-18T05:16:06.966Z] >           assert len(cpu) == len(gpu), "CPU and GPU list have different lengths at {} CPU: {} GPU: {}".format(path, len(cpu), len(gpu))

[2023-07-18T05:16:06.966Z] E           AssertionError: CPU and GPU list have different lengths at [0, 'sort_array(collect_set(b), true)'] CPU: 14 GPU: 20

[2023-07-18T05:16:06.966Z] 

[2023-07-18T05:16:06.966Z] ../../src/main/python/asserts.py:39: AssertionError

[2023-07-18T05:16:06.966Z] ----------------------------- Captured stdout call -----------------------------

[2023-07-18T05:16:06.966Z] ### CPU RUN ###

[2023-07-18T05:16:06.966Z] ### GPU RUN ###

[2023-07-18T05:16:06.966Z] ### COLLECT: GPU TOOK 0.19055843353271484 CPU TOOK 0.1242377758026123 ###

[2023-07-18T05:16:06.966Z] CPU OUTPUT: [Row(sort_array(collect_set(b), true)=[Row(child0=-110, child1=-11291, child2=358420087, child3=894377461322806935, child4=-1.6834864679694056e-31, child5=1.0303888663388996e-26, child6='¯óÏôÅkùê,dÌA~|×AÖ\x08VJé\x08yX^i{.\x1b\x83', child7=False, child8=datetime.date(4899, 3, 2), child9=datetime.datetime(6861, 9, 7, 19, 16, 50, 365146), child10=None), Row(child0=-108, child1=1267, child2=1917713880, child3=None, child4=None, child5=4.3214483342777574e-117, child6='[H\x89µ³hß\x14Æ\x12_XÕµoy\x08Y£àK;uk\x06\x15ç\x18\x02#', child7=True, child8=datetime.date(1590, 12, 27), child9=datetime.datetime(6166, 6, 5, 15, 21, 41, 824645), child10=None), Row(child0=-87, child1=18519, child2=105544272, child3=6446878210185090927, child4=None, child5=-4.2132496663975606e+247, child6='¦\r\x15\x16ep[|p\x9aË\x17ZZû\x82\x86\rê¼¨Ð³AúÍÿ\tÀò', child7=False, child8=datetime.date(3851, 2, 13), child9=datetime.datetime(297, 2, 6, 14, 13, 25, 35309), child10=None), Row(child0=-61, child1=-18968, child2=639916889, child3=2251193521986836658, child4=1.7728260685861134e+25, child5=-1.6653918844116466e+240, child6=' 7jgv#\x1aÑÚªï\x8d\x9fð\x15\x83\x1fîìAæöl\x0bå\\\x90æ9â', child7=True, child8=datetime.date(353, 7, 23), child9=datetime.datetime(551, 6, 24, 14, 12, 58, 187381), child10=None), Row(child0=-50, child1=-31138, child2=-1807864316, child3=-4041992126494898742, child4=4.065499056633676e+31, child5=7.554028173936504e-95, child6='ÆªOo©\x85·2Ôo!áP\x94\x9eþädÚ\x90õ4¢=L\x9däVi}', child7=True, child8=datetime.date(18, 7, 4), child9=datetime.datetime(6109, 8, 17, 22, 8, 3, 694767), child10=None), Row(child0=-41, child1=-24617, child2=848988934, child3=-7858370133784586516, child4=-1.740257146253725e-20, child5=3.5575370470599927e+284, child6='õ^\x1f·\x97ú·ÖF{/ZR*ø\x7fCýö\x05%A1Ð¶d\x04\x89øx', child7=True, child8=datetime.date(7802, 5, 13), child9=datetime.datetime(1279, 10, 18, 15, 49, 5, 515554), child10=None), Row(child0=-36, child1=28459, child2=2096953985, child3=1287002480498025462, child4=963.7640991210938, child5=9.376528689861087e+117, child6='\x05Â,ÀøYªe$«q;~\x04ëâ\x13h\x98ÇR\x04\x1e\x00©4@-\x0b°', child7=False, child8=datetime.date(8000, 2, 29), child9=datetime.datetime(5690, 4, 15, 13, 26, 32, 4190), child10=None), Row(child0=-27, child1=-24753, child2=-1726248225, child3=3307548583936611830, child4=-3.656386885506302e+18, child5=nan, child6='5\x92%w|ð\x9f\x98!\x887DÚdÌ$\x95Xò\xad\x98_ÿ>\x0b¡\tÇ(µ', child7=False, child8=datetime.date(9497, 10, 18), child9=datetime.datetime(8913, 8, 27, 22, 48, 29, 877849), child10=None), Row(child0=1, child1=-23550, child2=823668003, child3=-6580116972850632313, child4=3.816635312018178e+23, child5=8.600719030180411e+134, child6='\x0eM!R\x02ÇB\x1a¡^õ\x8cCÎµ\x1dpØX-Ùrz\x07\x9c¨\x1cÅ¨¡', child7=None, child8=datetime.date(5437, 1, 2), child9=None, child10=None), Row(child0=6, child1=None, child2=-394822741, child3=-6874717541235784212, child4=4243308150784.0, child5=-3.638751059169508e+285, child6='M\x80²ÝÄ\x81¬m]ø\x14å\x05L·\x99®\x8e\x9a\x1a}oÝ\x01à\nãÔ\x84\x08', child7=False, child8=datetime.date(7026, 5, 5), child9=datetime.datetime(860, 3, 3, 14, 32, 46, 946330), child10=None), Row(child0=39, child1=-22177, child2=347245469, child3=5615010007311415123, child4=-0.16667649149894714, child5=-2.1002678349193272e+151, child6="%{³ÁaÃ\x13ò£|dÐ'Y\x04B½u\x99èÈ(!Ú\x13,ÜhÔþ", child7=False, child8=datetime.date(8508, 12, 19), child9=datetime.datetime(7713, 4, 6, 2, 18, 9, 495678), child10=None), Row(child0=61, child1=-25185, child2=-2111287335, child3=-9223372036854775808, child4=None, child5=4.813757617752221e+57, child6=':\x9aLno»7þöµ\x00ý¦\x98vFWÊ\x9eeslr÷t±²ñ\x1eú', child7=True, child8=datetime.date(6996, 5, 13), child9=datetime.datetime(1691, 9, 13, 20, 49, 32, 239246), child10=None), Row(child0=84, child1=22049, child2=897674100, child3=7081246721334085708, child4=-5.445810266389943e+33, child5=-6.542279108216022e+206, child6='çìéô\x1cÀFS¥`\x04\x97ª¶\x14Ó\r¼\xa0¬ôÉe\x8dá~ì:£\x15', child7=False, child8=None, child9=datetime.datetime(2103, 9, 30, 11, 10, 53, 134710), child10=None), Row(child0=125, child1=9497, child2=2139679874, child3=7956328026836954420, child4=-243864059904.0, child5=2.1874890847202007e-206, child6='w\x96\x96\x1d\x8dÏ\x9b\x80\x86Úcy@5øäP\x86Ê\x81\x9coØr)\x8c{rY\x0c', child7=True, child8=datetime.date(3, 10, 7), child9=datetime.datetime(9259, 12, 8, 20, 39, 28, 751490), child10=None)])]

[2023-07-18T05:16:06.967Z] GPU OUTPUT: [Row(sort_array(collect_set(b), true)=[Row(child0=-110, child1=-11291, child2=358420087, child3=894377461322806935, child4=-1.6834864679694056e-31, child5=1.0303888663388996e-26, child6='¯óÏôÅkùê,dÌA~|×AÖ\x08VJé\x08yX^i{.\x1b\x83', child7=False, child8=datetime.date(4899, 3, 2), child9=datetime.datetime(6861, 9, 7, 19, 16, 50, 365146), child10=None), Row(child0=-108, child1=1267, child2=1917713880, child3=None, child4=None, child5=4.3214483342777574e-117, child6='[H\x89µ³hß\x14Æ\x12_XÕµoy\x08Y£àK;uk\x06\x15ç\x18\x02#', child7=True, child8=datetime.date(1590, 12, 27), child9=datetime.datetime(6166, 6, 5, 15, 21, 41, 824645), child10=None), Row(child0=-87, child1=18519, child2=105544272, child3=6446878210185090927, child4=None, child5=-4.2132496663975606e+247, child6='¦\r\x15\x16ep[|p\x9aË\x17ZZû\x82\x86\rê¼¨Ð³AúÍÿ\tÀò', child7=False, child8=datetime.date(3851, 2, 13), child9=datetime.datetime(297, 2, 6, 14, 13, 25, 35309), child10=None), Row(child0=-61, child1=-18968, child2=639916889, child3=2251193521986836658, child4=1.7728260685861134e+25, child5=-1.6653918844116466e+240, child6=' 7jgv#\x1aÑÚªï\x8d\x9fð\x15\x83\x1fîìAæöl\x0bå\\\x90æ9â', child7=True, child8=datetime.date(353, 7, 23), child9=datetime.datetime(551, 6, 24, 14, 12, 58, 187381), child10=None), Row(child0=-50, child1=-31138, child2=-1807864316, child3=-4041992126494898742, child4=4.065499056633676e+31, child5=7.554028173936504e-95, child6='ÆªOo©\x85·2Ôo!áP\x94\x9eþädÚ\x90õ4¢=L\x9däVi}', child7=True, child8=datetime.date(18, 7, 4), child9=datetime.datetime(6109, 8, 17, 22, 8, 3, 694767), child10=None), Row(child0=-41, child1=-24617, child2=848988934, child3=-7858370133784586516, child4=-1.740257146253725e-20, child5=3.5575370470599927e+284, child6='õ^\x1f·\x97ú·ÖF{/ZR*ø\x7fCýö\x05%A1Ð¶d\x04\x89øx', child7=True, child8=datetime.date(7802, 5, 13), child9=datetime.datetime(1279, 10, 18, 15, 49, 5, 515554), child10=None), Row(child0=-36, child1=28459, child2=2096953985, child3=1287002480498025462, child4=963.7640991210938, child5=9.376528689861087e+117, child6='\x05Â,ÀøYªe$«q;~\x04ëâ\x13h\x98ÇR\x04\x1e\x00©4@-\x0b°', child7=False, child8=datetime.date(8000, 2, 29), child9=datetime.datetime(5690, 4, 15, 13, 26, 32, 4190), child10=None), Row(child0=-27, child1=-24753, child2=-1726248225, child3=3307548583936611830, child4=-3.656386885506302e+18, child5=nan, child6='5\x92%w|ð\x9f\x98!\x887DÚdÌ$\x95Xò\xad\x98_ÿ>\x0b¡\tÇ(µ', child7=False, child8=datetime.date(9497, 10, 18), child9=datetime.datetime(8913, 8, 27, 22, 48, 29, 877849), child10=None), Row(child0=-27, child1=-24753, child2=-1726248225, child3=3307548583936611830, child4=-3.656386885506302e+18, child5=nan, child6='5\x92%w|ð\x9f\x98!\x887DÚdÌ$\x95Xò\xad\x98_ÿ>\x0b¡\tÇ(µ', child7=False, child8=datetime.date(9497, 10, 18), child9=datetime.datetime(8913, 8, 27, 22, 48, 29, 877849), child10=None), Row(child0=-27, child1=-24753, child2=-1726248225, child3=3307548583936611830, child4=-3.656386885506302e+18, child5=nan, child6='5\x92%w|ð\x9f\x98!\x887DÚdÌ$\x95Xò\xad\x98_ÿ>\x0b¡\tÇ(µ', child7=False, child8=datetime.date(9497, 10, 18), child9=datetime.datetime(8913, 8, 27, 22, 48, 29, 877849), child10=None), Row(child0=-27, child1=-24753, child2=-1726248225, child3=3307548583936611830, child4=-3.656386885506302e+18, child5=nan, child6='5\x92%w|ð\x9f\x98!\x887DÚdÌ$\x95Xò\xad\x98_ÿ>\x0b¡\tÇ(µ', child7=False, child8=datetime.date(9497, 10, 18), child9=datetime.datetime(8913, 8, 27, 22, 48, 29, 877849), child10=None), Row(child0=-27, child1=-24753, child2=-1726248225, child3=3307548583936611830, child4=-3.656386885506302e+18, child5=nan, child6='5\x92%w|ð\x9f\x98!\x887DÚdÌ$\x95Xò\xad\x98_ÿ>\x0b¡\tÇ(µ', child7=False, child8=datetime.date(9497, 10, 18), child9=datetime.datetime(8913, 8, 27, 22, 48, 29, 877849), child10=None), Row(child0=-27, child1=-24753, child2=-1726248225, child3=3307548583936611830, child4=-3.656386885506302e+18, child5=nan, child6='5\x92%w|ð\x9f\x98!\x887DÚdÌ$\x95Xò\xad\x98_ÿ>\x0b¡\tÇ(µ', child7=False, child8=datetime.date(9497, 10, 18), child9=datetime.datetime(8913, 8, 27, 22, 48, 29, 877849), child10=None), Row(child0=-27, child1=-24753, child2=-1726248225, child3=3307548583936611830, child4=-3.656386885506302e+18, child5=nan, child6='5\x92%w|ð\x9f\x98!\x887DÚdÌ$\x95Xò\xad\x98_ÿ>\x0b¡\tÇ(µ', child7=False, child8=datetime.date(9497, 10, 18), child9=datetime.datetime(8913, 8, 27, 22, 48, 29, 877849), child10=None), Row(child0=1, child1=-23550, child2=823668003, child3=-6580116972850632313, child4=3.816635312018178e+23, child5=8.600719030180411e+134, child6='\x0eM!R\x02ÇB\x1a¡^õ\x8cCÎµ\x1dpØX-Ùrz\x07\x9c¨\x1cÅ¨¡', child7=None, child8=datetime.date(5437, 1, 2), child9=None, child10=None), Row(child0=6, child1=None, child2=-394822741, child3=-6874717541235784212, child4=4243308150784.0, child5=-3.638751059169508e+285, child6='M\x80²ÝÄ\x81¬m]ø\x14å\x05L·\x99®\x8e\x9a\x1a}oÝ\x01à\nãÔ\x84\x08', child7=False, child8=datetime.date(7026, 5, 5), child9=datetime.datetime(860, 3, 3, 14, 32, 46, 946330), child10=None), Row(child0=39, child1=-22177, child2=347245469, child3=5615010007311415123, child4=-0.16667649149894714, child5=-2.1002678349193272e+151, child6="%{³ÁaÃ\x13ò£|dÐ'Y\x04B½u\x99èÈ(!Ú\x13,ÜhÔþ", child7=False, child8=datetime.date(8508, 12, 19), child9=datetime.datetime(7713, 4, 6, 2, 18, 9, 495678), child10=None), Row(child0=61, child1=-25185, child2=-2111287335, child3=-9223372036854775808, child4=None, child5=4.813757617752221e+57, child6=':\x9aLno»7þöµ\x00ý¦\x98vFWÊ\x9eeslr÷t±²ñ\x1eú', child7=True, child8=datetime.date(6996, 5, 13), child9=datetime.datetime(1691, 9, 13, 20, 49, 32, 239246), child10=None), Row(child0=84, child1=22049, child2=897674100, child3=7081246721334085708, child4=-5.445810266389943e+33, child5=-6.542279108216022e+206, child6='çìéô\x1cÀFS¥`\x04\x97ª¶\x14Ó\r¼\xa0¬ôÉe\x8dá~ì:£\x15', child7=False, child8=None, child9=datetime.datetime(2103, 9, 30, 11, 10, 53, 134710), child10=None), Row(child0=125, child1=9497, child2=2139679874, child3=7956328026836954420, child4=-243864059904.0, child5=2.1874890847202007e-206, child6='w\x96\x96\x1d\x8dÏ\x9b\x80\x86Úcy@5øäP\x86Ê\x81\x9coØr)\x8c{rY\x0c', child7=True, child8=datetime.date(3, 10, 7), child9=datetime.datetime(9259, 12, 8, 20, 39, 28, 751490), child10=None)])]

[2023-07-18T05:16:06.967Z] =============================== warnings summary ===============================

[2023-07-18T05:16:06.967Z] src/main/python/hash_aggregate_test.py: 32 warnings

[2023-07-18T05:16:06.967Z]   /spark-3.3.0-bin-hadoop3/python/pyspark/sql/functions.py:315: FutureWarning: Deprecated in 3.2, use sum_distinct instead.

[2023-07-18T05:16:06.967Z]     warnings.warn("Deprecated in 3.2, use sum_distinct instead.", FutureWarning)

[2023-07-18T05:16:06.967Z] 

[2023-07-18T05:16:06.967Z] src/main/python/hive_delimited_text_test.py: 56 warnings

[2023-07-18T05:16:06.967Z]   /spark-3.3.0-bin-hadoop3/python/pyspark/sql/catalog.py:309: FutureWarning: createExternalTable is deprecated since Spark 2.2, please use createTable instead.

[2023-07-18T05:16:06.967Z]     warnings.warn(

[2023-07-18T05:16:06.967Z] 

[2023-07-18T05:16:06.967Z] src/main/python/window_function_test.py::test_lru_cache_datagen[INJECT_OOM]

[2023-07-18T05:16:06.967Z]   /home/jenkins/agent/workspace/jenkins-JDK17-nightly-62/integration_tests/src/main/python/window_function_test.py:1452: UserWarning: Cache info: CacheInfo(hits=2580, misses=792, maxsize=128, currsize=128)

[2023-07-18T05:16:06.967Z]     warnings.warn("Cache info: {}".format(info))
```

--- a/docs/supported_ops.md
+++ b/docs/supported_ops.md
@@ -16595,9 +16595,9 @@ are limited.
 <td>S</td>
 <td><b>NS</b></td>
 <td><b>NS</b></td>
-<td><em>PS<br/>UTC is only supported TZ for child TIMESTAMP;<br/>unsupported child types BINARY, CALENDAR, MAP, UDT</em></td>
+<td><em>PS<br/>Arrays containing floats/doubles will not be supported.;<br/>UTC is only supported TZ for child TIMESTAMP;<br/>unsupported child types BINARY, CALENDAR, MAP, UDT</em></td>
 <td><b>NS</b></td>
-<td><em>PS<br/>UTC is only supported TZ for child TIMESTAMP;<br/>unsupported child types BINARY, CALENDAR, MAP, UDT</em></td>
+<td><em>PS<br/>Structs containing float/double will not be supported.;<br/>UTC is only supported TZ for child TIMESTAMP;<br/>unsupported child types BINARY, CALENDAR, MAP, UDT</em></td>
 <td><b>NS</b></td>
 </tr>
 <tr>
@@ -16638,9 +16638,9 @@ are limited.
 <td>S</td>
 <td><b>NS</b></td>
 <td><b>NS</b></td>
-<td><em>PS<br/>UTC is only supported TZ for child TIMESTAMP;<br/>unsupported child types BINARY, CALENDAR, MAP, UDT</em></td>
+<td><em>PS<br/>Arrays containing floats/doubles will not be supported.;<br/>UTC is only supported TZ for child TIMESTAMP;<br/>unsupported child types BINARY, CALENDAR, MAP, UDT</em></td>
 <td><b>NS</b></td>
-<td><em>PS<br/>UTC is only supported TZ for child TIMESTAMP;<br/>unsupported child types BINARY, CALENDAR, MAP, UDT</em></td>
+<td><em>PS<br/>Structs containing float/double will not be supported.;<br/>UTC is only supported TZ for child TIMESTAMP;<br/>unsupported child types BINARY, CALENDAR, MAP, UDT</em></td>
 <td><b>NS</b></td>
 </tr>
 <tr>
@@ -16681,9 +16681,9 @@ are limited.
 <td>S</td>
 <td><b>NS</b></td>
 <td><b>NS</b></td>
-<td><em>PS<br/>UTC is only supported TZ for child TIMESTAMP;<br/>unsupported child types BINARY, CALENDAR, MAP, UDT</em></td>
+<td><em>PS<br/>Arrays containing floats/doubles will not be supported.;<br/>UTC is only supported TZ for child TIMESTAMP;<br/>unsupported child types BINARY, CALENDAR, MAP, UDT</em></td>
 <td><b>NS</b></td>
-<td><em>PS<br/>UTC is only supported TZ for child TIMESTAMP;<br/>unsupported child types BINARY, CALENDAR, MAP, UDT</em></td>
+<td><em>PS<br/>Structs containing float/double will not be supported.;<br/>UTC is only supported TZ for child TIMESTAMP;<br/>unsupported child types BINARY, CALENDAR, MAP, UDT</em></td>
 <td><b>NS</b></td>
 </tr>
 <tr>

--- a/integration_tests/src/main/python/data_gen.py
+++ b/integration_tests/src/main/python/data_gen.py
@@ -1013,6 +1013,9 @@ all_basic_gens = all_basic_gens_no_null + [null_gen]
 all_basic_gens_no_nan = [byte_gen, short_gen, int_gen, long_gen, FloatGen(no_nans=True), DoubleGen(no_nans=True),
         string_gen, boolean_gen, date_gen, timestamp_gen, null_gen]
 
+all_basic_gens_no_floats = [byte_gen, short_gen, int_gen, long_gen,
+        string_gen, boolean_gen, date_gen, timestamp_gen, null_gen]
+
 # Many Spark versions have issues sorting large decimals,
 # see https://issues.apache.org/jira/browse/SPARK-40089.
 orderable_decimal_gen_128bit = decimal_gen_128bit
@@ -1045,6 +1048,8 @@ single_level_array_gens_no_null = [ArrayGen(sub_gen) for sub_gen in all_basic_ge
 
 single_level_array_gens_no_nan = [ArrayGen(sub_gen) for sub_gen in all_basic_gens_no_nan + decimal_gens]
 
+single_level_array_gens_no_floats = [ArrayGen(sub_gen) for sub_gen in all_basic_gens_no_floats + decimal_gens]
+
 single_level_array_gens_no_decimal = [ArrayGen(sub_gen) for sub_gen in all_basic_gens]
 
 map_string_string_gen = [MapGen(StringGen(pattern='key_[0-9]', nullable=False), StringGen())]
@@ -1063,7 +1068,13 @@ all_basic_struct_gen = StructGen([['child'+str(ind), sub_gen] for ind, sub_gen i
 
 all_basic_struct_gen_no_nan = StructGen([['child'+str(ind), sub_gen] for ind, sub_gen in enumerate(all_basic_gens_no_nan)])
 
-struct_array_gen_no_nans =  StructGen([['child'+str(ind), sub_gen] for ind, sub_gen in enumerate(single_level_array_gens_no_nan)])
+all_basic_struct_gen_no_floats = StructGen([['child'+str(ind), sub_gen] for ind, sub_gen in enumerate(all_basic_gens_no_floats)])
+
+struct_array_gen_no_nans = StructGen([['child'+str(ind), sub_gen] for ind, sub_gen in enumerate(single_level_array_gens_no_nan)])
+
+struct_array_gen_no_floats = StructGen([['child'+str(ind), sub_gen] for ind, sub_gen in enumerate(single_level_array_gens_no_floats)])
+
+struct_array_gen = StructGen([['child'+str(ind), sub_gen] for ind, sub_gen in enumerate(single_level_array_gens_no_nan)])
 
 # Some struct gens, but not all because of nesting
 nonempty_struct_gens_sample = [all_basic_struct_gen,

--- a/integration_tests/src/main/python/data_gen.py
+++ b/integration_tests/src/main/python/data_gen.py
@@ -1070,11 +1070,11 @@ all_basic_struct_gen_no_nan = StructGen([['child'+str(ind), sub_gen] for ind, su
 
 all_basic_struct_gen_no_floats = StructGen([['child'+str(ind), sub_gen] for ind, sub_gen in enumerate(all_basic_gens_no_floats)])
 
+struct_array_gen = StructGen([['child'+str(ind), sub_gen] for ind, sub_gen in enumerate(single_level_array_gens)])
+
 struct_array_gen_no_nans = StructGen([['child'+str(ind), sub_gen] for ind, sub_gen in enumerate(single_level_array_gens_no_nan)])
 
 struct_array_gen_no_floats = StructGen([['child'+str(ind), sub_gen] for ind, sub_gen in enumerate(single_level_array_gens_no_floats)])
-
-struct_array_gen = StructGen([['child'+str(ind), sub_gen] for ind, sub_gen in enumerate(single_level_array_gens_no_nan)])
 
 # Some struct gens, but not all because of nesting
 nonempty_struct_gens_sample = [all_basic_struct_gen,

--- a/integration_tests/src/main/python/hash_aggregate_test.py
+++ b/integration_tests/src/main/python/hash_aggregate_test.py
@@ -624,6 +624,10 @@ _gen_data_for_collect_set_op = [[
     ('a', RepeatSeqGen(LongGen(), length=20)),
     ('b', value_gen)] for value_gen in _repeat_agg_column_for_collect_set_op]
 
+_gen_data_for_collect_set_op_floats = [[
+    ('a', RepeatSeqGen(LongGen(), length=20)),
+    ('b', RepeatSeqGen(struct_array_gen, length=15))]]
+
 _gen_data_for_collect_set_op_nested = [[
     ('a', RepeatSeqGen(LongGen(), length=20)),
     ('b', value_gen)] for value_gen in _repeat_agg_column_for_collect_set_op_nested + _array_of_array_gen]
@@ -697,6 +701,15 @@ def test_hash_groupby_collect_set_on_nested_type(data_gen):
         lambda spark: gen_df(spark, data_gen, length=100)
             .groupby('a')
             .agg(f.sort_array(f.collect_set('b'))))
+    
+@ignore_order(local=True)
+@allow_non_gpu('ObjectHashAggregateExec', 'ShuffleExchangeExec', 'CollectSet')
+@pytest.mark.parametrize('data_gen', _gen_data_for_collect_set_op_floats, ids=idfn)
+def test_hash_groupby_collect_set_fallback_on_nested_floats(data_gen):
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark: gen_df(spark, data_gen, length=100)
+            .groupby('a')
+            .agg(f.sort_array(f.collect_set('b'))))
 
 
 # Note, using sort_array() on the CPU, because sort_array() does not yet
@@ -739,6 +752,13 @@ def test_hash_reduction_collect_set_on_nested_type(data_gen):
         lambda spark: gen_df(spark, data_gen, length=100)
             .agg(f.sort_array(f.collect_set('b'))))
 
+@ignore_order(local=True)
+@allow_non_gpu('ObjectHashAggregateExec', 'ShuffleExchangeExec', 'CollectSet')
+@pytest.mark.parametrize('data_gen', _gen_data_for_collect_set_op_floats, ids=idfn)
+def test_hash_reduction_collect_set_fallback_on_nested_floats(data_gen):
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark: gen_df(spark, data_gen, length=100)
+            .agg(f.sort_array(f.collect_set('b'))))
 
 # Note, using sort_array() on the CPU, because sort_array() does not yet
 # support sorting certain nested/arbitrary types on the GPU

--- a/integration_tests/src/main/python/hash_aggregate_test.py
+++ b/integration_tests/src/main/python/hash_aggregate_test.py
@@ -621,13 +621,13 @@ _repeat_agg_column_for_collect_set_op_nested = [
 struct_array_gen_floats = StructGen([['child'+str(ind), sub_gen] for ind, sub_gen 
                                         in enumerate([ArrayGen(double_gen), ArrayGen(float_gen)])])
 
-floats_struct_gen = StructGen([['child'+str(ind), sub_gen] for ind, sub_gen in enumerate([float_gen, double_gen])])
+struct_gen_floats = StructGen([['child'+str(ind), sub_gen] for ind, sub_gen in enumerate([float_gen, double_gen])])
 
 _repeat_agg_column_for_collect_set_op_nested_floats = [
     RepeatSeqGen(struct_array_gen_floats, length=15),
     RepeatSeqGen(StructGen([
         ['c0', struct_array_gen_floats], ['c1', int_gen]]), length=15),
-    RepeatSeqGen(ArrayGen(floats_struct_gen), length=15),
+    RepeatSeqGen(ArrayGen(struct_gen_floats), length=15),
     RepeatSeqGen(ArrayGen(ArrayGen(float_gen)), length=15),
     RepeatSeqGen(ArrayGen(ArrayGen(double_gen)), length=15)]
 

--- a/integration_tests/src/main/python/hash_aggregate_test.py
+++ b/integration_tests/src/main/python/hash_aggregate_test.py
@@ -701,15 +701,19 @@ def test_hash_groupby_collect_set_on_nested_type(data_gen):
         lambda spark: gen_df(spark, data_gen, length=100)
             .groupby('a')
             .agg(f.sort_array(f.collect_set('b'))))
-    
+
+# Fall back to CPU if type is nested and contains floats or doubles in the type tree.
+# Because NaNs in nested types are not supported yet. 
+# See https://github.com/NVIDIA/spark-rapids/issues/8808   
 @ignore_order(local=True)
 @allow_non_gpu('ObjectHashAggregateExec', 'ShuffleExchangeExec', 'CollectSet')
 @pytest.mark.parametrize('data_gen', _gen_data_for_collect_set_op_floats, ids=idfn)
 def test_hash_groupby_collect_set_fallback_on_nested_floats(data_gen):
-    assert_gpu_and_cpu_are_equal_collect(
+    assert_gpu_fallback_collect(
         lambda spark: gen_df(spark, data_gen, length=100)
             .groupby('a')
-            .agg(f.sort_array(f.collect_set('b'))))
+            .agg(f.sort_array(f.collect_set('b'))),
+            'CollectSet')
 
 
 # Note, using sort_array() on the CPU, because sort_array() does not yet
@@ -752,13 +756,17 @@ def test_hash_reduction_collect_set_on_nested_type(data_gen):
         lambda spark: gen_df(spark, data_gen, length=100)
             .agg(f.sort_array(f.collect_set('b'))))
 
+# Fall back to CPU if type is nested and contains floats or doubles in the type tree.
+# Because NaNs in nested types are not supported yet. 
+# See https://github.com/NVIDIA/spark-rapids/issues/8808
 @ignore_order(local=True)
 @allow_non_gpu('ObjectHashAggregateExec', 'ShuffleExchangeExec', 'CollectSet')
 @pytest.mark.parametrize('data_gen', _gen_data_for_collect_set_op_floats, ids=idfn)
 def test_hash_reduction_collect_set_fallback_on_nested_floats(data_gen):
-    assert_gpu_and_cpu_are_equal_collect(
+    assert_gpu_fallback_collect(
         lambda spark: gen_df(spark, data_gen, length=100)
-            .agg(f.sort_array(f.collect_set('b'))))
+            .agg(f.sort_array(f.collect_set('b'))),
+            'CollectSet')
 
 # Note, using sort_array() on the CPU, because sort_array() does not yet
 # support sorting certain nested/arbitrary types on the GPU

--- a/integration_tests/src/main/python/hash_aggregate_test.py
+++ b/integration_tests/src/main/python/hash_aggregate_test.py
@@ -618,6 +618,19 @@ _repeat_agg_column_for_collect_set_op_nested = [
         ['c0', struct_array_gen_no_floats], ['c1', int_gen]]), length=15),
     RepeatSeqGen(ArrayGen(all_basic_struct_gen_no_floats), length=15)]
 
+struct_array_gen_floats = StructGen([['child'+str(ind), sub_gen] for ind, sub_gen 
+                                        in enumerate([ArrayGen(double_gen), ArrayGen(float_gen)])])
+
+floats_struct_gen = StructGen([['child'+str(ind), sub_gen] for ind, sub_gen in enumerate([float_gen, double_gen])])
+
+_repeat_agg_column_for_collect_set_op_nested_floats = [
+    RepeatSeqGen(struct_array_gen_floats, length=15),
+    RepeatSeqGen(StructGen([
+        ['c0', struct_array_gen_floats], ['c1', int_gen]]), length=15),
+    RepeatSeqGen(ArrayGen(floats_struct_gen), length=15),
+    RepeatSeqGen(ArrayGen(ArrayGen(float_gen)), length=15),
+    RepeatSeqGen(ArrayGen(ArrayGen(double_gen)), length=15)]
+
 _array_of_array_gen = [RepeatSeqGen(ArrayGen(sub_gen), length=15) for sub_gen in single_level_array_gens_no_floats]
 
 _gen_data_for_collect_set_op = [[
@@ -631,6 +644,10 @@ _gen_data_for_collect_set_op_floats = [[
 _gen_data_for_collect_set_op_nested = [[
     ('a', RepeatSeqGen(LongGen(), length=20)),
     ('b', value_gen)] for value_gen in _repeat_agg_column_for_collect_set_op_nested + _array_of_array_gen]
+
+_gen_data_for_collect_set_op_nested_floats = [[
+    ('a', RepeatSeqGen(LongGen(), length=20)),
+    ('b', value_gen)] for value_gen in _repeat_agg_column_for_collect_set_op_nested_floats]
 
 _all_basic_gens_with_all_nans_cases = all_basic_gens + [SetValuesGen(t, [math.nan, None]) for t in [FloatType(), DoubleType()]]
 
@@ -707,7 +724,8 @@ def test_hash_groupby_collect_set_on_nested_type(data_gen):
 # See https://github.com/NVIDIA/spark-rapids/issues/8808   
 @ignore_order(local=True)
 @allow_non_gpu('ObjectHashAggregateExec', 'ShuffleExchangeExec', 'CollectSet')
-@pytest.mark.parametrize('data_gen', _gen_data_for_collect_set_op_floats, ids=idfn)
+@pytest.mark.parametrize('data_gen', _gen_data_for_collect_set_op_floats + 
+                         _gen_data_for_collect_set_op_nested_floats, ids=idfn)
 def test_hash_groupby_collect_set_fallback_on_nested_floats(data_gen):
     assert_gpu_fallback_collect(
         lambda spark: gen_df(spark, data_gen, length=100)
@@ -761,7 +779,7 @@ def test_hash_reduction_collect_set_on_nested_type(data_gen):
 # See https://github.com/NVIDIA/spark-rapids/issues/8808
 @ignore_order(local=True)
 @allow_non_gpu('ObjectHashAggregateExec', 'ShuffleExchangeExec', 'CollectSet')
-@pytest.mark.parametrize('data_gen', _gen_data_for_collect_set_op_floats, ids=idfn)
+@pytest.mark.parametrize('data_gen', _gen_data_for_collect_set_op_floats + _gen_data_for_collect_set_op_nested_floats, ids=idfn)
 def test_hash_reduction_collect_set_fallback_on_nested_floats(data_gen):
     assert_gpu_fallback_collect(
         lambda spark: gen_df(spark, data_gen, length=100)

--- a/integration_tests/src/main/python/hash_aggregate_test.py
+++ b/integration_tests/src/main/python/hash_aggregate_test.py
@@ -603,18 +603,22 @@ _gen_data_for_collect_list_op = _full_gen_data_for_collect_op + [[
     ('b', value_gen)] for value_gen in _repeat_agg_column_for_collect_list_op]
 
 _repeat_agg_column_for_collect_set_op = [
-    RepeatSeqGen(all_basic_struct_gen_no_nan, length=15),
+    RepeatSeqGen(all_basic_struct_gen_no_floats, length=15),
     RepeatSeqGen(StructGen([
-        ['c0', all_basic_struct_gen_no_nan], ['c1', int_gen]]), length=15)]
+        ['c0', all_basic_struct_gen_no_floats], ['c1', int_gen]]), length=15)]
+
+struct_array_gen_no_nans = StructGen([['child'+str(ind), sub_gen] for ind, sub_gen in enumerate(single_level_array_gens_no_nan)])
+
+struct_array_gen_no_floats = StructGen([['child'+str(ind), sub_gen] for ind, sub_gen in enumerate(single_level_array_gens_no_floats)])
 
 # data generating for collect_set based-nested Struct[Array] types
 _repeat_agg_column_for_collect_set_op_nested = [
-    RepeatSeqGen(struct_array_gen_no_nans, length=15),
+    RepeatSeqGen(struct_array_gen_no_floats, length=15),
     RepeatSeqGen(StructGen([
-        ['c0', struct_array_gen_no_nans], ['c1', int_gen]]), length=15),
-    RepeatSeqGen(ArrayGen(all_basic_struct_gen_no_nan), length=15)]
+        ['c0', struct_array_gen_no_floats], ['c1', int_gen]]), length=15),
+    RepeatSeqGen(ArrayGen(all_basic_struct_gen_no_floats), length=15)]
 
-_array_of_array_gen = [RepeatSeqGen(ArrayGen(sub_gen), length=15) for sub_gen in single_level_array_gens_no_nan]
+_array_of_array_gen = [RepeatSeqGen(ArrayGen(sub_gen), length=15) for sub_gen in single_level_array_gens_no_floats]
 
 _gen_data_for_collect_set_op = [[
     ('a', RepeatSeqGen(LongGen(), length=20)),
@@ -1175,8 +1179,8 @@ _no_neg_zero_all_basic_gens = [byte_gen, short_gen, int_gen, long_gen,
         FloatGen(special_cases=[FLOAT_MIN, FLOAT_MAX, 0.0, 1.0, -1.0]), DoubleGen(special_cases=[]),
         string_gen, boolean_gen, date_gen, timestamp_gen]
 
-_struct_only_nested_gens = [all_basic_struct_gen,
-                            StructGen([['child0', byte_gen], ['child1', all_basic_struct_gen]]),
+_struct_only_nested_gens = [all_basic_struct_gen_no_floats,
+                            StructGen([['child0', byte_gen], ['child1', all_basic_struct_gen_no_floats]]),
                             StructGen([])]
 @pytest.mark.parametrize('data_gen',
                          _no_neg_zero_all_basic_gens + decimal_gens + _struct_only_nested_gens,

--- a/integration_tests/src/main/python/hash_aggregate_test.py
+++ b/integration_tests/src/main/python/hash_aggregate_test.py
@@ -603,9 +603,9 @@ _gen_data_for_collect_list_op = _full_gen_data_for_collect_op + [[
     ('b', value_gen)] for value_gen in _repeat_agg_column_for_collect_list_op]
 
 _repeat_agg_column_for_collect_set_op = [
-    RepeatSeqGen(all_basic_struct_gen, length=15),
+    RepeatSeqGen(all_basic_struct_gen_no_nan, length=15),
     RepeatSeqGen(StructGen([
-        ['c0', all_basic_struct_gen], ['c1', int_gen]]), length=15)]
+        ['c0', all_basic_struct_gen_no_nan], ['c1', int_gen]]), length=15)]
 
 # data generating for collect_set based-nested Struct[Array] types
 _repeat_agg_column_for_collect_set_op_nested = [
@@ -688,7 +688,6 @@ def test_hash_groupby_collect_set(data_gen):
 
 @ignore_order(local=True)
 @pytest.mark.parametrize('data_gen', _gen_data_for_collect_set_op, ids=idfn)
-@pytest.mark.xfail(reason='https://github.com/NVIDIA/spark-rapids/issues/8716')
 def test_hash_groupby_collect_set_on_nested_type(data_gen):
     assert_gpu_and_cpu_are_equal_collect(
         lambda spark: gen_df(spark, data_gen, length=100)
@@ -731,7 +730,6 @@ def test_hash_reduction_collect_set(data_gen):
 
 @ignore_order(local=True)
 @pytest.mark.parametrize('data_gen', _gen_data_for_collect_set_op, ids=idfn)
-@pytest.mark.xfail(reason='https://github.com/NVIDIA/spark-rapids/issues/8716')
 def test_hash_reduction_collect_set_on_nested_type(data_gen):
     assert_gpu_and_cpu_are_equal_collect(
         lambda spark: gen_df(spark, data_gen, length=100)

--- a/integration_tests/src/main/python/window_function_test.py
+++ b/integration_tests/src/main/python/window_function_test.py
@@ -1137,6 +1137,14 @@ _gen_data_for_collect_set_nested = [
     ('c_array_array_decimal_128', RepeatSeqGen(ArrayGen(ArrayGen(decimal_gen_128bit)), length=15)),
 ]
 
+_gen_data_for_collect_set_nest_floats_fallback = [
+    ('a', RepeatSeqGen(LongGen(), length=20)),
+    ('b', UniqueLongGen()),
+    ('c_int', RepeatSeqGen(IntegerGen(), length=15)),
+    ('c_array_array_float', RepeatSeqGen(ArrayGen(ArrayGen(FloatGen(no_nans=True))), length=15)),
+    ('c_array_array_double', RepeatSeqGen(ArrayGen(ArrayGen(DoubleGen(no_nans=True))), length=15)),
+]
+
 # SortExec does not support array type, so sort the result locally.
 @ignore_order(local=True)
 def test_window_aggs_for_rows_collect_set():
@@ -1265,6 +1273,30 @@ def test_window_aggs_for_rows_collect_set_nested_array():
         """)
     assert_gpu_and_cpu_are_equal_collect(do_it, conf=conf)
 
+# Fall back to CPU if type is nested and contains floats or doubles in the type tree.
+# Because NaNs in nested types are not supported yet. 
+# See https://github.com/NVIDIA/spark-rapids/issues/8808
+@ignore_order(local=True)
+@allow_non_gpu("ProjectExec", "CollectSet", "WindowExec")
+def test_window_aggs_for_rows_collect_set_nested_array():
+    conf = copy_and_update(_float_conf, {
+        "spark.rapids.sql.castFloatToString.enabled": "true",
+        "spark.rapids.sql.expression.SortArray": "false"
+    })
+
+    def do_it(spark):
+        df = gen_df(spark, _gen_data_for_collect_set_nest_floats_fallback, length=512)
+        df.createOrReplaceTempView("window_collect_table")
+        return spark.sql(
+            """select a, b,
+              collect_set(c_array_array_float) over
+                (partition by a order by b,c_int rows between CURRENT ROW and UNBOUNDED FOLLOWING) as cc_array_array_float,
+              collect_set(c_array_array_double) over
+                (partition by a order by b,c_int rows between CURRENT ROW and UNBOUNDED FOLLOWING) as cc_array_array_double
+        from window_collect_table
+        """)
+        
+    assert_gpu_fallback_collect(do_it, 'CollectSet', conf=conf)
 
 # In a distributed setup the order of the partitions returned might be different, so we must ignore the order
 # but small batch sizes can make sort very slow, so do the final order by locally

--- a/integration_tests/src/main/python/window_function_test.py
+++ b/integration_tests/src/main/python/window_function_test.py
@@ -1120,10 +1120,10 @@ _gen_data_for_collect_set_nested = [
     ('a', RepeatSeqGen(LongGen(), length=20)),
     ('b', UniqueLongGen()),
     ('c_int', RepeatSeqGen(IntegerGen(), length=15)),
-    ('c_struct_array_1', RepeatSeqGen(struct_array_gen_no_nans, length=15)),
+    ('c_struct_array_1', RepeatSeqGen(struct_array_gen_no_floats, length=15)),
     ('c_struct_array_2', RepeatSeqGen(StructGen([
-        ['c0', struct_array_gen_no_nans], ['c1', int_gen]]), length=14)),
-    ('c_array_struct', RepeatSeqGen(ArrayGen(all_basic_struct_gen_no_nan), length=15)),
+        ['c0', struct_array_gen_no_floats], ['c1', int_gen]]), length=14)),
+    ('c_array_struct', RepeatSeqGen(ArrayGen(all_basic_struct_gen_no_floats), length=15)),
     ('c_array_array_bool', RepeatSeqGen(ArrayGen(ArrayGen(BooleanGen())), length=15)),
     ('c_array_array_int', RepeatSeqGen(ArrayGen(ArrayGen(IntegerGen())), length=15)),
     ('c_array_array_long', RepeatSeqGen(ArrayGen(ArrayGen(LongGen())), length=15)),
@@ -1132,8 +1132,6 @@ _gen_data_for_collect_set_nested = [
     ('c_array_array_timestamp', RepeatSeqGen(ArrayGen(ArrayGen(TimestampGen())), length=15)),
     ('c_array_array_byte', RepeatSeqGen(ArrayGen(ArrayGen(ByteGen())), length=15)),
     ('c_array_array_string', RepeatSeqGen(ArrayGen(ArrayGen(StringGen())), length=15)),
-    ('c_array_array_float', RepeatSeqGen(ArrayGen(ArrayGen(FloatGen(no_nans=True))), length=15)),
-    ('c_array_array_double', RepeatSeqGen(ArrayGen(ArrayGen(DoubleGen(no_nans=True))), length=15)),
     ('c_array_array_decimal_32', RepeatSeqGen(ArrayGen(ArrayGen(DecimalGen(precision=8, scale=3))), length=15)),
     ('c_array_array_decimal_64', RepeatSeqGen(ArrayGen(ArrayGen(decimal_gen_64bit)), length=15)),
     ('c_array_array_decimal_128', RepeatSeqGen(ArrayGen(ArrayGen(decimal_gen_128bit)), length=15)),
@@ -1235,10 +1233,6 @@ def test_window_aggs_for_rows_collect_set_nested_array():
                 (partition by a order by b,c_int rows between CURRENT ROW and UNBOUNDED FOLLOWING) as cc_array_array_byte,
               collect_set(c_array_array_string) over
                 (partition by a order by b,c_int rows between CURRENT ROW and UNBOUNDED FOLLOWING) as cc_array_array_str,
-              collect_set(c_array_array_float) over
-                (partition by a order by b,c_int rows between CURRENT ROW and UNBOUNDED FOLLOWING) as cc_array_array_float,
-              collect_set(c_array_array_double) over
-                (partition by a order by b,c_int rows between CURRENT ROW and UNBOUNDED FOLLOWING) as cc_array_array_double,
               collect_set(c_array_array_decimal_32) over
                 (partition by a order by b,c_int rows between CURRENT ROW and UNBOUNDED FOLLOWING) as cc_array_array_decimal_32,
               collect_set(c_array_array_decimal_64) over
@@ -1264,8 +1258,6 @@ def test_window_aggs_for_rows_collect_set_nested_array():
               sort_array(cc_array_array_ts),
               sort_array(cc_array_array_byte),
               sort_array(cc_array_array_str),
-              sort_array(cc_array_array_float),
-              sort_array(cc_array_array_double),
               sort_array(cc_array_array_decimal_32),
               sort_array(cc_array_array_decimal_64),
               sort_array(cc_array_array_decimal_128)

--- a/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala
+++ b/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala
@@ -3260,9 +3260,9 @@ object GpuOverrides extends Logging {
           (TypeSig.commonCudfTypes + TypeSig.DECIMAL_128 +
               TypeSig.NULL +
               TypeSig.STRUCT.withPsNote(TypeEnum.STRUCT, "Structs containing " +
-              s"float/double array will not be supported.") +
+              s"float/double will not be supported.") +
               TypeSig.ARRAY.withPsNote(TypeEnum.ARRAY, "Arrays containing " +
-              s"floats/doubles array will not be supported.")).nested(),
+              s"floats/doubles will not be supported.")).nested(),
           TypeSig.all))),
       (c, conf, p, r) => new TypedImperativeAggExprMeta[CollectSet](c, conf, p, r) {
 

--- a/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala
+++ b/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala
@@ -3264,18 +3264,13 @@ object GpuOverrides extends Logging {
           TypeSig.all))),
       (c, conf, p, r) => new TypedImperativeAggExprMeta[CollectSet](c, conf, p, r) {
 
-        private def isNestedArrayType(dt: DataType): Boolean = {
-          dt match {
-            case StructType(fields) =>
-              fields.exists { field =>
-                field.dataType match {
-                  case sdt: StructType => isNestedArrayType(sdt)
-                  case _: ArrayType => true
-                  case _ => false
-                }
-              }
-            case ArrayType(et, _) => et.isInstanceOf[ArrayType] || et.isInstanceOf[StructType]
-            case _ => false
+        override def tagAggForGpu(): Unit = {
+          // Fall back to CPU if type is nested and contains floats or doubles in the type tree.
+          // Because NaNs in nested types are not supported yet. 
+          // See https://github.com/NVIDIA/spark-rapids/issues/8808
+          if (c.child.dataType != FloatType && c.child.dataType != DoubleType &&
+              isOrContainsFloatingPoint(c.child.dataType)) {
+            willNotWorkOnGpu("Float/Double in nested type is not supported")
           }
         }
 

--- a/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala
+++ b/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala
@@ -3259,8 +3259,10 @@ object GpuOverrides extends Logging {
         Seq(ParamCheck("input",
           (TypeSig.commonCudfTypes + TypeSig.DECIMAL_128 +
               TypeSig.NULL +
-              TypeSig.STRUCT +
-              TypeSig.ARRAY).nested(),
+              TypeSig.STRUCT.withPsNote(TypeEnum.STRUCT, "Structs containing " +
+              s"float/double array will not be supported.") +
+              TypeSig.ARRAY.withPsNote(TypeEnum.ARRAY, "Arrays containing " +
+              s"floats/doubles array will not be supported.")).nested(),
           TypeSig.all))),
       (c, conf, p, r) => new TypedImperativeAggExprMeta[CollectSet](c, conf, p, r) {
 

